---
title: "Comp_Hym_Proj_Jul22_2"
output: html_document
date: "2022-07-14"
---

14th July 2022

Gene level differential expression analysis.

```{r}
set.seed(42)
```

##Libraries

```{r}
library("DESeq2")
library("tidyverse")
library("ggrepel")
library("tximport")
library("pheatmap")
library("reshape")
library("ggvenn")
```

##Resources

Following recommended workflow: http://master.bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html#preparing-quantification-input-to-deseq2

DeSeq2: https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html

Trim Galore:  https://github.com/FelixKrueger/TrimGalore

Kallisto :    https://pachterlab.github.io/kallisto/about

Gene level instructions from tximport: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4712774/

OrthoGroup information / Annotation information

```{r}
ortho <- read.table("input/Ortho_Files/OrthoFinder_Combined_Parsed_Jun22.tsv",
                    sep = "\t", header = T)

ann <- read.table("input/Gene_Info/Amel_ImmAnnotations_Orthogroup_Jun22.tsv",
                  header = T, sep = "\t")
ann.slim <- ann[,c(2:3)]
```

##Functions

Converts transcript-level counts from input/ directory to a single counts table with gene level estimates. Also generates a coldata object in the output directory

```{r}
geneCount <- function(species){
  dir <- paste("input/Counts_Kal/", species, sep = "")
  files <- list.files(dir)
  samples <- sapply(strsplit(files, ".", fixed = T), "[", 1)
  treat <- c(rep("Naive", 3), rep("Wound", 3), rep("GramPos", 3),
             rep("GramNeg",3))
  coldata <- as.data.frame(cbind(samples, treat))
  names(coldata) <- c("Sample", "Treatment")
  write.table(coldata, 
              paste("output/Ind_DE/", species, "/", species, "_SampleData.tsv", sep = ""),
              col.names = T, row.names = F, sep = "\t", quote = F)
  names(files) <- samples
  txgene <- read.table(paste("input/Gene_Info/", species, "_TXGene.tsv", sep =""),
                       header = T, sep = "\t")
  cnts <- file.path(dir, files)
  names(cnts) <- samples
  txi.kal.tsv <- tximport(cnts, type = "kallisto", tx2gene = txgene, ignoreAfterBar = T)
  write.table(txi.kal.tsv$counts,
              paste("output/Ind_DE/", species, "/", 
                    species,  "_TxGeneCount.tsv", sep = ""),
              col.names = T, row.names = T, quote = F, sep = "\t")
  return(txi.kal.tsv)
}
```

Produce DDS object from tximport object (includes prefiltering and setting reference condition to Naive)

```{r}
getDDS <- function(species, txobject){
  cd <- read.table(paste("output/Ind_DE/", species, "/",
                         species, "_SampleData.tsv", sep = ""), header = T, sep = "\t")
ddsTx <- DESeqDataSetFromTximport(txobject, colData = cd, design = ~Treatment)
keep <- rowSums(counts(ddsTx)) >= 10
ddsTx2 <- ddsTx[keep,]
ddsTx2$Treatment <- relevel(ddsTx2$Treatment, ref = "Naive")
ddsTx2 <- DESeq(ddsTx2)
  return(ddsTx2)
}
```

Visually assess the samples if (an) outlier(s) was/were removed

```{r}
outCheck <- function(outlier_s, species){
  cd <- read.table(paste("output/Ind_DE/", species, "/",
                         species, "_SampleData.tsv", sep = ""), header = T)
  cd <- cd[!cd$Sample %in% outlier_s,]
  dir <- paste("input/Counts_Kal/", species, sep = "")
  files <- list.files(dir)
  samples <- sapply(strsplit(files, ".", fixed = T), "[", 1)
  txgene <- read.table(paste("input/Gene_Info/", species, "_TXGene.tsv", sep =""),
                         header = T, sep = "\t")
  cnts <- file.path(dir, files)
  names(cnts) <- samples
  cnts[outlier_s]
  cnts <- cnts[!names(cnts) %in% outlier_s]
  tx <- tximport(cnts, type = "kallisto", tx2gene = txgene, ignoreAfterBar = T)
  
  ddsTx <- DESeqDataSetFromTximport(tx, colData = cd, design = ~Treatment)
  keep <- rowSums(counts(ddsTx)) >= 10
  ddsTx2 <- ddsTx[keep,]
  ddsTx2$Treatment <- relevel(ddsTx2$Treatment, ref = "Naive")
  ddsTx2 <- DESeq(ddsTx2)
  vsd <- vst(ddsTx2, blind = F)
  select <- order(rowMeans(counts(ddsTx2,normalized=TRUE)),
                decreasing=TRUE)[1:20]
  p1 <- pheatmap(assay(vsd)[select,])
  sampleDists <- dist(t(assay(vsd)))
  sampleDistMatrix <- as.matrix(sampleDists)
  rownames(sampleDistMatrix) <- paste(vsd$Treatment, vsd$type, sep="-")
  colnames(sampleDistMatrix) <- paste(vsd$Treatment, vsd$type, sep="-")
  p2 <- pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)
  pca <- plotPCA(vsd, intgroup="Treatment", returnData = TRUE)
  p3 <- ggplot(pca, aes(x = PC1, y = PC2, color = Treatment, label = name)) +
    geom_point() +
    geom_text()
  p1
  p2 
  p3
}
```

Return a dds object with specified outlier removed

```{r}
ddsOmit <- function(outlier_s, species){
 cd <- read.table(paste("output/Ind_DE/", species, "/",
                         species, "_SampleData.tsv", sep = ""), header = T)
  cd <- cd[!cd$Sample %in% outlier_s,]
  dir <- paste("input/Counts_Kal/", species, sep = "")
  files <- list.files(dir)
  samples <- sapply(strsplit(files, ".", fixed = T), "[", 1)
  txgene <- read.table(paste("input/Gene_Info/", species, "_TXGene.tsv", sep =""),
                         header = T, sep = "\t")
  cnts <- file.path(dir, files)
  names(cnts) <- samples
  cnts[outlier_s]
  cnts <- cnts[!names(cnts) %in% outlier_s]
  tx <- tximport(cnts, type = "kallisto", tx2gene = txgene, ignoreAfterBar = T)
  
  ddsTx <- DESeqDataSetFromTximport(tx, colData = cd, design = ~Treatment)
  keep <- rowSums(counts(ddsTx)) >= 10
  ddsTx2 <- ddsTx[keep,]
  ddsTx2$Treatment <- relevel(ddsTx2$Treatment, ref = "Naive")
  ddsTx2 <- DESeq(ddsTx2)
  return(ddsTx2)
}
```

Produce a dataframe with orthogroup and gene class information ready for data visualisation

```{r}
makeMetaDf <- function(res.all.object){
  x <- res.all.object[!is.na(res.all.object$padj),]
  x$Condition[x$Contrast==paste(treats[1])] <- "Wound"
  x$Condition[x$Contrast==paste(treats[2])] <- "Gram Positive"
  x$Condition[x$Contrast==paste(treats[3])] <- "Gram Negative"
  for(i in 1:nrow(x)){
  check <- x$Gene[i] %in% ortho$GeneID
    if (check == FALSE){
      x$OrthoGroup[i] <- "Unassigned"
    } else {
    x$OrthoGroup[i] <- ortho$Orthogroup[ortho$GeneID ==
                                                       x$Gene[i]]
    }
  }
  for(i in 1:nrow(x)){
  if (x$OrthoGroup[i] == "Unassigned") {
    x$Function[i] <- "Unknown"} else {
      check2 <- x$OrthoGroup[i] %in% ann.slim$Orthogroup
      if (check2 == FALSE){
        x$Function[i] <- "Unknown"
      } else {
    f <- unique(ann.slim$Function[ann.slim$Orthogroup == x$OrthoGroup[i]])
    if (length(f) > 1){
      if(TRUE %in% grepl("Effector|Recognition|Signalling", f)){
        x$Function[i] <- "Immune"
    } else {
      if(TRUE %in% grepl("Putative Immune", f)){
        x$Function[i] <- "Putative Immune"
        } 
      }
    } else {
    f2 <- gsub("Effector", "Immune", 
               gsub("Recognition", "Immune", 
                    gsub("Signalling", "Immune", 
                         gsub("Background", "None Immune", f))))
    x$Function[i] <- paste(f2)
        }
      }
    }
  }
  x$Sig <- "No"
  x$Sig[x$padj < 0.1] <- "Yes"
  x$Condition <- factor(x$Condition, levels = c("Wound", 
                                                "Gram Negative", 
                                                "Gram Positive"))
  x$Neglog10adjPvalue <- -log10(x$padj)
  x$Direction <- "Not Significant"
  x$Direction[x$Sig == "Yes" &
                    x$log2FoldChange > 0 ] <- "Up-Regulated"
  x$Direction[x$Sig == "Yes" &
                    x$log2FoldChange < 0 ] <- "Down-Regulated"
  return(x)
}
```

Produce a 1/0 binary matrix for use in comparing overlap in genes of interest

```{r}
makeMatrix <- function(goidf){
  mat <- as.data.frame(unique(goidf$goi))
  names(mat) <- "Gene"
  mat$SigWound <- ifelse(mat$Gene %in% sig.wound, 1, 0)
  mat$SigPos <- ifelse(mat$Gene %in% sig.pos, 1, 0)
  mat$SigNeg <- ifelse(mat$Gene %in% sig.neg, 1, 0)
  mat$TopWound <- ifelse(mat$Gene %in% top.wound, 1, 0)
  mat$TopPos <- ifelse(mat$Gene %in% top.pos, 1, 0)
  mat$TopNeg <- ifelse(mat$Gene %in% top.neg, 1, 0)
  mat$AllSig <- ifelse(mat$SigWound == 1 &
                      mat$SigPos == 1 &
                      mat$SigNeg == 1, 1, 0)
  mat$AllTop <- ifelse(mat$TopWound == 1 &
                      mat$TopPos == 1 &
                      mat$TopNeg == 1, 1, 0)
  mat$BacterialSig <- ifelse(mat$SigWound == 0 &
                      mat$SigPos == 1 &
                      mat$SigNeg == 1, 1, 0)
  mat$BacterialTop <- ifelse(mat$TopWound == 0 &
                      mat$TopPos == 1 &
                      mat$TopNeg == 1, 1, 0)
  mat$GoiWound <- ifelse(mat$TopWound == 1 | mat$SigWound == 1, 1, 0)
  mat$GoiPos <- ifelse(mat$TopPos == 1 | mat$SigPos == 1, 1, 0)
  mat$GoiNeg <- ifelse(mat$TopNeg == 1 | mat$SigNeg == 1, 1, 0)
  mat$AllGoi <- ifelse(mat$GoiWound == 1 &
                      mat$GoiPos == 1 &
                        mat$GoiNeg == 1, 1, 0)
  mat$BacterialGoi <- ifelse(mat$GoiPos == 1 &
                        mat$GoiNeg == 1  &
                          mat$GoiWound == 0, 1, 0)
  return(mat)
}
```

Produce and save a boxplot with expression levels across samples with gene input

```{r}
visiGene <- function(gene, genename, dds){
  tmp <- counts(dds, normalized = T)
  samples <- dds$Sample
  counts <- tmp[rownames(tmp) == paste(gene),]
  plot.df <- as.data.frame(cbind(samples, counts))
  names(plot.df) <- c("Samples", "Norm_Counts")
  plot.df$Treatment[grepl("_N_", plot.df$Samples)] <- "Naive"
  plot.df$Treatment[grepl("_P_", plot.df$Samples)] <- "Wound"
  plot.df$Treatment[grepl("_SM_", plot.df$Samples)] <- "Gram Negative"
  plot.df$Treatment[grepl("_SL_", plot.df$Samples)] <- "Gram Positive"
  plot.df$Treatment <- factor(plot.df$Treatment, levels = c("Naive",
                                                        "Wound", 
                                                        "Gram Positive", 
                                                        "Gram Negative"))
  plot.df$Norm_Counts <- as.numeric(plot.df$Norm_Counts)
  p <- ggplot(plot.df, aes(x = Treatment, y = log(Norm_Counts), fill = Treatment)) +
    geom_boxplot() +
    geom_point(position = position_jitter(w=0.1, h = 0),
               aes(colour = Treatment), size = 3) +
    scale_colour_manual(values = c("#58ffa7", "#ffa758", "#ff58b0", "#58b0ff")) +
    xlab("Treatment") +
    labs(title = (paste(genename, " expression across treatments", sep = ""))) +
    guides(alpha = FALSE, colour = guide_legend("Treatment"))
  ggsave(paste(wd, "Plots/ExpressionPlots/", gene, "_ExpBoxPlot.pdf", sep = ""))
  p
}
```

Plot all expression boxplots from a group of genes 

```{r}
visiFacet <- function(genes, genenames, dds, filename){
  p <- vector(mode = "list", length = length(genes))
  for (i in 1:length(p)){
    tmp <- counts(dds, normalized = T)
    samples <- dds$Sample
    counts <- tmp[rownames(tmp) == paste(genes[i]),]
    p[[i]] <- as.data.frame(cbind(samples, counts))
    names(p[[i]]) <- c("Samples", "Norm_Counts")
    p[[i]]$Treatment[grepl("_N_", p[[i]]$Samples)] <- "Naive"
    p[[i]]$Treatment[grepl("_P_", p[[i]]$Samples)] <- "Wound"
    p[[i]]$Treatment[grepl("_SM_", p[[i]]$Samples)] <- "Gram Negative"
    p[[i]]$Treatment[grepl("_SL_", p[[i]]$Samples)] <- "Gram Positive"
    p[[i]]$Treatment <- factor(p[[i]]$Treatment, levels = c("Naive",
                                                          "Wound", 
                                                          "Gram Positive", 
                                                          "Gram Negative"))
    p[[i]]$Norm_Counts <- as.numeric(p[[i]]$Norm_Counts)
    p[[i]]$Gene <- paste(genes[i])
    p[[i]]$Label <- paste(genenames[i])
  }
  p2 <- bind_rows(p)
  p3 <- ggplot(p2, aes(x = Treatment, y = log(Norm_Counts))) +
    geom_boxplot(aes(colour = Treatment, alpha = 0.5)) +
    geom_point(position = position_jitter(w=0.1, h = 0),
               aes(colour = Treatment), size = 3) +
    scale_colour_manual(values = c("#58ffa7", "#ffa758", "#ff58b0", "#58b0ff")) +
    xlab("Treatment") +
    guides(alpha = FALSE, colour = guide_legend("Treatment")) +
    facet_wrap(~Label, nrow = 5, scale = "free_y")
  ggsave(paste(wd, "Plots/ExpressionPlots/", filename, "_Facet_ExpBoxPlot.pdf", sep = ""))
  p3
}
```



##Caus: Gene Level DE 

```{r}
dir.create("output/Ind_DE/Caus/")
wd <- "output/Ind_DE/Caus/"
```

###DE Analysis

First, convert transcript-level counts table from Kallisto to gene level counts and produce a coldata object

```{r}
caus.tx <- geneCount("Caus")
head(caus.tx$counts)
```

Build a DESeq DataSet from the above, prefilter (keep only rows that have at least 10 reads total), and return dds object.

```{r}
caus.dds <- getDDS("Caus", caus.tx)
caus.res <- vector(mode = "list", length = 3)
treats <- resultsNames(caus.dds)[c(4,3,2)]

for(i in 1:length(caus.res)){
  caus.res[[i]] <- results(caus.dds, name = paste(treats[i]))
  summary(caus.res[[i]])
}
```

###Check for outliers

First, check counts

```{r}
boxplot(log10(assays(caus.dds)[["cooks"]]), range=0, las=2)
```

```{r}
#normalise
vsd <- vst(caus.dds, blind = F)
#make heatmap (top 20 genes)
select <- order(rowMeans(counts(caus.dds,normalized=TRUE)),
                decreasing=TRUE)[1:20]
pheatmap(assay(vsd)[select,])
```
This is messy I'm not sure where to begin.

```{r}
pdf("output/Ind_DE/Caus/Caus_Pheatmap.pdf")
  pheatmap(assay(vsd)[select,])
dev.off()
```

Check with sample to sample distances

```{r}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$Treatment, vsd$type, sep="-")
colnames(sampleDistMatrix) <- NULL

pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)
```
Yeah that's pretty messy
```{r}
pdf("output/Ind_DE/Caus/Caus_DistanceMatrix.pdf")
  pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)
dev.off()
```

Finally, PCA

```{r}
caus.pca <- plotPCA(vsd, intgroup="Treatment", returnData = TRUE)
ggplot(caus.pca, aes(x = PC1, y = PC2, color = Treatment, label = name)) +
  geom_point() +
  geom_text()
```
```{r}
ggsave("output/Ind_DE/Caus/Caus_PCA.pdf")
```

P55 and SL56 seem a bit ... away
What happens if I remove the potential outliers ? 

```{r}
torem <- c("CA_P_055", "CA_SL_056", "CA_N_012")
outCheck(torem, "Caus")
```

2 plots are still a mess, but the sample-distance plots are very much improved.
Now N 12 is a problem ? Go back and try again....

Much better.... though I've lost 3 samples!


```{r}
caus.dds2 <- ddsOmit(torem, "Caus")

caus.res2 <- vector(mode = "list", length = 3)
treats2 <- resultsNames(caus.dds2)[c(4,3,2)]

for(i in 1:length(caus.res2)){
  caus.res2[[i]] <- results(caus.dds2, name = paste(treats2[i]))
  print(treats2[i])
  summary(caus.res2[[i]])
}
```

Save plots

```{r}
vsd2 <- vst(caus.dds2, blind = F)
#make heatmap (top 20 genes)
select <- order(rowMeans(counts(caus.dds2,normalized=TRUE)),
                decreasing=TRUE)[1:20]
pheatmap(assay(vsd2)[select,])
```

```{r}
pdf("output/Ind_DE/Caus/Caus_OutlierRemoved_Pheatmap.pdf")
  pheatmap(assay(vsd2)[select,])
dev.off()
```

```{r}
sampleDists2 <- dist(t(assay(vsd2)))
sampleDistMatrix2 <- as.matrix(sampleDists2)
rownames(sampleDistMatrix2) <- paste(vsd2$Treatment, vsd2$type, sep="-")
colnames(sampleDistMatrix2) <- NULL

pheatmap(sampleDistMatrix2,
         clustering_distance_rows=sampleDists2,
         clustering_distance_cols=sampleDists2)
```

```{r}
pdf("output/Ind_DE/Caus/Caus_OutlierRemoved_DistanceMatrix.pdf")
  pheatmap(sampleDistMatrix2,
         clustering_distance_rows=sampleDists2,
         clustering_distance_cols=sampleDists2)
dev.off()
```


```{r}
caus.pca2 <- plotPCA(vsd2, intgroup="Treatment", returnData = TRUE)
ggplot(caus.pca2, aes(x = PC1, y = PC2, color = Treatment, label = name)) +
  geom_point() +
  geom_text()
ggsave("output/Ind_DE/Caus/Caus_OutlierRemoved_PCA.pdf")
```

Do we see a difference in number of sig genes in terms of whether they're immune or not ? 

```{r}
sig <- vector(mode = "list", length = length(caus.res))

for (i in 1:length(sig)){
  tmp <- caus.res[[i]][!is.na(caus.res),]
  tmp$Gene <- rownames(tmp)
  sig[[i]] <- tmp$Gene[tmp$padj < 0.1]
  sig[[i]] <- sig[[i]][!is.na(sig[[i]])]
}

run1 <- unique(unlist(sig))
ort1 <- vector(length = length(run1))
ann1 <- vector(length = length(run1))

for(i in 1:length(run1)){
  check <- run1[i] %in% ortho$GeneID
  if (check == FALSE){
  ort1[i] <- "Unassigned"
  ann1[i] <- "Unknown"
} else {
  ort1[i] <- ortho$Orthogroup[ortho$GeneID == run1[i]]
  check2 <- ort1[i] %in% ann.slim$Orthogroup
  if (check2 == FALSE){
    ann1[i] <- "Unknown"
  } else {
  f <- unique(ann.slim$Function[ann.slim$Orthogroup == ort1[i]])
  if (length(f) > 1){
    if(TRUE %in% grepl("Effector|Recognition|Signalling", f)){
      ann1[i] <- "Immune"
    } else {
        if(TRUE %in% grepl("Putative Immune", f)){
          ann1[i] <- "Putative Immune"
        }
      }
  } else {
      f2 <- gsub("Effector", "Immune", 
               gsub("Recognition", "Immune", 
                    gsub("Signalling", "Immune", 
                         gsub("Background", "None Immune", f))))
      ann1[i] <- paste(f2)
      }
  }
  }
}

one <- as.data.frame(cbind(run1, ort1, ann1))
names(one) <- c("Gene", "OrthoGroup", "Function")
one$Run <- paste("One")
```

And again...

```{r}
sig2 <- vector(mode = "list", length = length(caus.res2))

for (i in 1:length(sig2)){
  tmp <- caus.res2[[i]][!is.na(caus.res2),]
  tmp$Gene <- rownames(tmp)
  sig2[[i]] <- tmp$Gene[tmp$padj < 0.1]
  sig2[[i]] <- sig2[[i]][!is.na(sig2[[i]])]
}

run2 <- unique(unlist(sig2))
ort2 <- vector(length = length(run2))
ann2 <- vector(length = length(run2))

for(i in 1:length(run2)){
  check <- run2[i] %in% ortho$GeneID
  if (check == FALSE){
  ort2[i] <- "Unassigned"
  ann2[i] <- "Unknown"
} else {
  ort2[i] <- ortho$Orthogroup[ortho$GeneID == run2[i]]
  check2 <- ort2[i] %in% ann.slim$Orthogroup
  if (check2 == FALSE){
    ann2[i] <- "Unknown"
  } else {
  f <- unique(ann.slim$Function[ann.slim$Orthogroup == ort2[i]])
  if (length(f) > 1){
    if(TRUE %in% grepl("Effector|Recognition|Signalling", f)){
      ann2[i] <- "Immune"
    } else {
        if(TRUE %in% grepl("Putative Immune", f)){
          ann2[i] <- "Putative Immune"
        }
      }
  } else {
      f2 <- gsub("Effector", "Immune", 
               gsub("Recognition", "Immune", 
                    gsub("Signalling", "Immune", 
                         gsub("Background", "None Immune", f))))
      ann2[i] <- paste(f2)
      }
  }
  }
}


two <- as.data.frame(cbind(run2, ort2, ann2))
names(two) <- c("Gene", "OrthoGroup", "Function")
two$Run <- paste("Two")
```


```{r}
one <- as.data.frame(table(one$Function, one$Run))
one$Prop <- one$Freq/sum(one$Freq)
two <- as.data.frame(table(two$Function, two$Run))
two$Prop <- two$Freq/sum(two$Freq)

check.df <- rbind(one,two)

ggplot(check.df, aes(fill=Var1, y=Prop, x=Var2)) + 
    geom_bar(position="dodge", stat="identity")
```

```{r}
ggplot(check.df, aes(fill=Var1, y=Freq, x=Var2)) + 
    geom_bar(position="dodge", stat="identity")
```

There is quite a significant difference .... gonna make the call to go forward without the outliers but .... this may change with Seth's input.

```{r}
caus.dds <- caus.dds2
caus.res <- caus.res2
```


###Results

Save 

```{r}
for (i in 1:length(caus.res)){
  caus.res[[i]]$Contrast <- paste(treats[i])
  caus.res[[i]]$Gene <- rownames(caus.res[[i]])
  rownames(caus.res[[i]]) <- NULL
  caus.res[[i]] <- as.data.frame(caus.res[[i]])
}

caus.res.all <- bind_rows(caus.res[1:3])

id <- read.table("input/Gene_Info/Caus_IDwCcal.tsv", 
                 sep = "\t", quote = "", header = T)

for (i in 1:nrow(caus.res.all)){
  d <- unique(id$CcalID[id$Gene == caus.res.all$Gene[i]])
  if(length(d) > 1){
    caus.res.all$Desc[i] <- d[1]
  } else {
    caus.res.all$Desc[i] <- d  
  }
}

write.table(caus.res.all, paste(wd, "Caus_DeSeqResults_All.tsv", sep = ""),
            col.names = T, row.names = F, quote = F, sep = "\t")

caus.res.sig <- subset(caus.res.all, padj < 0.1)

write.table(caus.res.sig, paste(wd, "Caus_DeSeqResults_FDRlt0.1.tsv", sep =""), 
            col.names = T, row.names = F, quote = F, sep = "\t")
```

Genes of interest?

I'm going to take the top 10 most significant and top 10 biggest logFC per treatment. 
Genes that overlap between two (and between the three treatments) will be of particular interest

```{r}
nona <- caus.res.all[!is.na(caus.res.all$padj),]
nona$LogDif <- sub('-', '', nona$log2FoldChange)
tmp <- nona[nona$Contrast == treats[1],]
sig.wound <- head(tmp$Gene[order(tmp$padj)], n = 10)
top.wound <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)
tmp <- nona[nona$Contrast == treats[2],]
sig.pos <- head(tmp$Gene[order(tmp$padj)], n = 10)
top.pos <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)
tmp <- nona[nona$Contrast == treats[3],]
sig.neg <-  head(tmp$Gene[order(tmp$padj)], n = 10)
top.neg <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)

goi <- c(sig.wound, top.wound, sig.pos, top.pos, sig.neg, top.neg)
stat <- c(rep("Top 10 Sig Wound", 10), rep("Top 10 log2FC Wound", 10),
          rep("Top 10 Sig Gram Positive", 10), rep("Top 10 log2FC Gram Positive", 10),
          rep("Top 10 Sig Gram Negative", 10), rep("Top 10 log2FC Gram Negative", 10))

goi.df <- as.data.frame(cbind(goi, stat))


goi.wound <- unique(c(sig.wound, top.wound))
goi.pos <- unique(c(sig.pos, top.pos))
goi.neg <- unique(c(sig.neg, top.neg))
```

###Visualise: Venn Diagrams

```{r}
top.sig <- list("Wound" = sig.wound,
                 "Gram Positive" = sig.pos,
                 "Gram Negative" = sig.neg)

p <- ggvenn(top.sig,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Top 10 Genes per Treatment by adjPvalue")
```

```{r}
dir.create(paste(wd, "Plots/", sep = ""))
ggsave(paste(wd, "Plots/Caus_TopAdjPvalue_Venn.pdf", sep = ""))
```

```{r}
top.lfc <- list("Wound" = top.wound,
                 "Gram Positive" = top.pos,
                 "Gram Negative" = top.neg)

p <- ggvenn(top.lfc,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Top 10 Genes per Treatment by log2FC")
```



```{r}
ggsave(paste(wd, "Plots/Caus_Toplog2FC_Venn.pdf", sep = ""))
```

```{r}
goi <- list("Wound" = goi.wound,
                 "Gram Positive" = goi.pos,
                 "Gram Negative" = goi.neg)

p <- ggvenn(goi,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Genes of Interest Across Treatments")
```


```{r}
ggsave(paste(wd, "Plots/Caus_GOI_Venn.pdf", sep = ""))
```

```{r}
goi.mat <- makeMatrix(goi.df)
goi.mat
```

```{r}
goi.mat$Gene[goi.mat$AllGoi == 1]
```

uncharacterized LOC108629120, homeobox protein 2-like, cytochrome P450 4g15-like, uncharacterized LOC108627515, transcription factor TFIIIB component B'' homolog, peroxisome assembly protein 12

```{r}
goi.mat$Gene[goi.mat$BacterialGoi == 1]
```

uncharacterized LOC108629123, cysteine sulfinic acid decarboxylase, origin recognition complex subunit 4

```{r}
goi.mat$Gene[goi.mat$AllSig == 1]
```

uncharacterized LOC108629120, homeobox protein 2-like, cytochrome P450 4g15-like

```{r}
goi.mat$Gene[goi.mat$AllTop == 1]
```

uncharacterised LOC108627515, transcription factor TFIIIB component B'' homolog, peroxisome assembly protein 12

####Visualise: Expression BoxPlots

```{r}
dir.create(paste(wd, "Plots/ExpressionPlots/", sep = ""))
```

####Significant Wound

```{r}
goi.mat$Gene[goi.mat$SigWound == 1]
genes <- goi.mat$Gene[goi.mat$SigWound == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = caus.dds)
}
```


All Wound: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = caus.dds, 
          filename = "WoundSig")
```


####TopFC Wound


```{r}
goi.mat$Gene[goi.mat$TopWound == 1]
genes <- goi.mat$Gene[goi.mat$TopWound == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = caus.dds)
}
```


All Wound: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = caus.dds, 
          filename = "Woundlog2FC")
```


####Significant GramPos


```{r}
goi.mat$Gene[goi.mat$SigPos == 1]
genes <- goi.mat$Gene[goi.mat$SigPos == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = caus.dds)
}
```

All Gram Pos Sig: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = caus.dds, 
          filename = "PosSig")
```


####TopFC GramPos

```{r}
goi.mat$Gene[goi.mat$TopPos == 1]
genes <- goi.mat$Gene[goi.mat$TopPos == 1]
```


```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = caus.dds)
}
```


All log Gram Pos: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = caus.dds, 
          filename = "Poslog2FC")
```

####Significant GramNeg


```{r}
goi.mat$Gene[goi.mat$SigNeg == 1]
genes <- goi.mat$Gene[goi.mat$SigNeg == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = caus.dds)
}
```


All Gram Neg Sig: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = caus.dds, 
          filename = "NegSig")
```


####TopFC GramNeg


```{r}
goi.mat$Gene[goi.mat$TopNeg == 1]
genes <- goi.mat$Gene[goi.mat$TopNeg == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = caus.dds)
}
```

All log Gram Neg: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = caus.dds, 
          filename = "Neglog2FC")
```


###Visualise: PCA

Redo after removing outliers.

```{r}
vsd <- vst(caus.dds, blind = F)

caus.pca <- plotPCA(vsd, intgroup="Treatment", returnData = TRUE)

ggplot(caus.pca, aes(x = PC1, y = PC2, color = Treatment)) +
  geom_point() 
```

```{r}
ggsave(paste(wd, "Plots/Caus_PCA_Simple.pdf", sep = ""))
```

```{r}
ggplot(caus.pca, aes(x = PC1, y = PC2, color = Treatment)) +
  geom_point() +
  geom_label_repel(
    data = caus.pca,
    aes(label = name),
    size = 3
  )
```


```{r}
ggsave(paste(wd, "Plots/Caus_PCA_Labelled.pdf", sep = ""))
```

```{r}
caus.pca$Treat2 <- c(rep("Naive", 2), rep("Treatment", 7))

ggplot(caus.pca, aes(x = PC1, y = PC2, color = Treat2)) +
  geom_point() +
  stat_ellipse()
```

```{r}
ggsave(paste(wd, "Plots/Caus_PCA_TrtVUnTrt.pdf", sep = ""))
```

###Visualise: Smears

```{r}
plot.df <- makeMetaDf(caus.res.all)
plot.df %>%
  filter(Condition == "Wound") %>%
  arrange(padj)
```


```{r}
plot.df$Condition <- factor(plot.df$Condition, levels = c("Wound",
                                                          "Gram Positive",
                                                          "Gram Negative"))

ggplot(plot.df, aes(x = log(baseMean), y = log2FoldChange)) +
  geom_point(aes(colour = Sig)) +
  facet_grid(~Condition) +
  scale_color_manual(values=c("black", "Red"), guide = NULL) +
  theme(legend.position = "bottom")
```



```{r}
ggsave(paste(wd, "Plots/Caus_SmearPlot.pdf", sep = ""))
```


###Visualise: Rainclouds

```{r}
sig.genes <- unique(plot.df$Gene[plot.df$padj < 0.1])
plot.sig <- plot.df[plot.df$Gene %in% sig.genes,]

to.add.naive <- plot.sig[plot.sig$Gene %in% sig.genes &
                               plot.sig$Condition == "Wound",]
to.add.naive$Sig <- "No"
to.add.naive$Condition <- "Naive"
to.add.naive$log2FoldChange <- 0

plot.sig <- rbind(plot.sig, to.add.naive)

plot.sig$Condition <- factor(plot.sig$Condition, levels = c("Naive",
                                                            "Wound", 
                                                            "Gram Positive", 
                                                            "Gram Negative"))

plot.sig$Significance[plot.sig$padj > 0.1] <- "Not Significant"
plot.sig$Significance[plot.sig$padj < 0.1 & plot.sig$padj > 0.05] <- 
  "Significant (FDR < 0.1)"
plot.sig$Significance[plot.sig$padj < 0.05 & plot.sig$padj > 0.01] <-
  "Significant (FDR < 0.05)"
plot.sig$Significance[plot.sig$padj < 0.01 & plot.sig$padj > 0.001] <-
  "Significant (FDR < 0.01)"
plot.sig$Significance[plot.sig$padj < 0.001] <-
  "Significant (FDR < 0.001)"

plot.sig$Significance <- factor(plot.sig$Significance, levels =
                                 c("Not Significant",
                                   "Significant (FDR < 0.1)",
                                   "Significant (FDR < 0.05)",
                                   "Significant (FDR < 0.01)",
                                   "Significant (FDR < 0.001)"))


plot.sig$Function <- factor(plot.sig$Function, levels = c("Immune",
                                                          "Putative Immune", 
                                                          "None Immune", 
                                                          "Unknown"))

ggplot(plot.sig[!plot.sig$Condition == "Naive",], 
       (aes(x = Condition, y = log2FoldChange))) +
  geom_jitter(aes(colour = Function, shape = Significance)) +
  scale_colour_manual(values = c("#2c8cd9", "#7a2cd9", "darkgrey", "plum")) +
  scale_shape_manual(values = c(1, 5, 18, 19, 17)) 
```

```{r}
ggsave(paste(wd, "Plots/Caus_logFCJitter.pdf", sep = ""))
```

```{r}
ggplot(plot.sig, 
       (aes(x = Condition, y = log2FoldChange))) +
  geom_point(aes(colour = Function, shape = Significance)) +
  geom_line( linetype = "dashed", alpha = 0.2, aes(group = Gene, colour = Function)) +
  scale_colour_manual(values = c("#2c8cd9", "#7a2cd9", "darkgrey", "plum")) +
  scale_shape_manual(values = c(1, 5, 18, 19, 17)) 
```

```{r}
ggsave(paste(wd, "Plots/Caus_logFCScatter_GeneLine.pdf", sep = ""))
```

####Visualise: Volcano Plots

```{r}
plot.df$Function <- factor(plot.df$Function, levels = c("Immune",
                                                        "Putative Immune",
                                                        "None Immune",
                                                        "Unknown"))

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.1", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Caus_Simple_FDR0.1_Volcano.pdf", sep = ""))
```


```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.1", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```
```{r}
ggsave(paste(wd, "Plots/Caus_Class_FDR0.1_Volcano.pdf", sep = ""))
```

```{r}
plot.df$Direction2 <- "Not Significant"
plot.df$Direction2[plot.df$padj < 0.05 &
                    plot.df$log2FoldChange > 0 ] <- "Up-Regulated"
plot.df$Direction2[plot.df$padj < 0.05 &
                    plot.df$log2FoldChange < 0 ] <- "Down-Regulated"

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction2), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.05", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```

```{r}
ggsave(paste(wd, "Plots/Caus_Simple_FDR0.05_Volcano.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction2, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.05", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Caus_Class_FDR0.05_Volcano.pdf", sep = ""))
```


```{r}
plot.df$Direction3 <- "Not Significant"
plot.df$Direction3[plot.df$padj < 0.001 &
                    plot.df$log2FoldChange > 0 ] <- "Up-Regulated"
plot.df$Direction3[plot.df$padj < 0.001 &
                    plot.df$log2FoldChange < 0 ] <- "Down-Regulated"

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Caus_Simple_FDR0.001_Volcano.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```

```{r}
ggsave(paste(wd, "Plots/Caus_Class_FDR0.001_Volcano.pdf", sep = ""))
```

Label top 3 sig genes per treatment

```{r}
sig.wound[1:3]
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "Caust.v2_016347"] <- "ATP-citrate synthase	"
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "Caust.v2_020967"] <- "Piwi-like protein Siwi"
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "Caust.v2_011640"] <- "TIM50-like"

sig.pos[1:3]
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "Caust.v2_018203"] <- "NT5D3-like"
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "Caust.v2_011034"] <- "Uncharacterised LOC108629120"
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "Caust.v2_011210"] <- "Med24"

sig.neg[1:3]
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "Caust.v2_011640"] <- "TIM50-like"
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "Caust.v2_011034"] <- "Uncharacterised LOC108629120"
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "Caust.v2_011550"] <- "Uncharacterised LOC108623077"
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") +
  geom_label_repel(
    data = plot.df[!is.na(plot.df$Label),],
    size = 2,
    aes(label = Label),
    point.padding = unit(0.3, "lines"),
    box.padding = 1
  ) 

```


```{r}
ggsave(paste(wd, "Plots/Caus_Simple_FDR0.001_Volcano_top3Label.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") +
  geom_label_repel(
    data = plot.df[!is.na(plot.df$Label),],
    size = 2,
    aes(label = Label),
    point.padding = unit(0.3, "lines"),
    box.padding = 2.25
  ) 

```
```{r}
ggsave(paste(wd, "Plots/Caus_Class_FDR0.001_Volcano_top3Label.pdf", sep = ""))
```


##Plan: Gene Level DE 

```{r}
dir.create("output/Ind_DE/Plan/")
wd <- "output/Ind_DE/Plan/"
```

###DE Analysis

First, convert transcript-level counts table from Kallisto to gene level counts and produce a coldata object

```{r}
plan.tx <- geneCount("Plan")
head(plan.tx$counts)
```

Build a DESeq DataSet from the above, prefilter (keep only rows that have at least 10 reads total), and return dds object.

```{r}
plan.dds <- getDDS("plan", plan.tx)
plan.res <- vector(mode = "list", length = 3)
treats <- resultsNames(plan.dds)[c(4,3,2)]

for(i in 1:length(plan.res)){
  plan.res[[i]] <- results(plan.dds, name = paste(treats[i]))
  summary(plan.res[[i]])
}
```

###Check for outliers

First, check counts

```{r}
boxplot(log10(assays(plan.dds)[["cooks"]]), range=0, las=2)
```

```{r}
#normalise
vsd <- vst(plan.dds, blind = F)
#make heatmap (top 20 genes)
select <- order(rowMeans(counts(plan.dds,normalized=TRUE)),
                decreasing=TRUE)[1:20]
pheatmap(assay(vsd)[select,])
```
N1908 may be an issue.

```{r}
pdf("output/Ind_DE/Plan/Plan_Pheatmap.pdf")
  pheatmap(assay(vsd)[select,])
dev.off()
```

Check with sample to sample distances

```{r}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$Treatment, vsd$type, sep="-")
colnames(sampleDistMatrix) <- NULL

pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)
```
Yeah that's pretty messy

```{r}
pdf("output/Ind_DE/Plan/Plan_DistanceMatrix.pdf")
  pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)
dev.off()
```

Finally, PCA

```{r}
plan.pca <- plotPCA(vsd, intgroup="Treatment", returnData = TRUE)
ggplot(plan.pca, aes(x = PC1, y = PC2, color = Treatment, label = name)) +
  geom_point() +
  geom_text()
```
And N1909 ...
I can't remove two from the same treatment, I need an absolute minimum of 2.

```{r}
ggsave("output/Ind_DE/Plan/Plan_PCA.pdf")
```

Going with 1908 and SM1936
If I could remove 1909 as well, I would
What happens if I remove the potential outliers ? 


```{r}
torem <- c("PL_N_1908", "PL_SM_1936", "PL_P_1912")
outCheck(torem, "Plan")
```

2 plots are still a mess, but the sample-distance plots are very much improved.
Hmm. P1912....


```{r}
plan.dds2 <- ddsOmit(torem, "Plan")

plan.res2 <- vector(mode = "list", length = 3)
treats2 <- resultsNames(plan.dds2)[c(4,3,2)]

for(i in 1:length(plan.res2)){
  plan.res2[[i]] <- results(plan.dds2, name = paste(treats2[i]))
  print(treats2[i])
  summary(plan.res2[[i]])
}
```

Wow. That's a lot of genes.

Save plots

```{r}
vsd2 <- vst(plan.dds2, blind = F)
#make heatmap (top 20 genes)
select <- order(rowMeans(counts(plan.dds2,normalized=TRUE)),
                decreasing=TRUE)[1:20]
pheatmap(assay(vsd2)[select,])
```

```{r}
pdf("output/Ind_DE/Plan/Plan_OutlierRemoved_Pheatmap.pdf")
  pheatmap(assay(vsd2)[select,])
dev.off()
```

```{r}
sampleDists2 <- dist(t(assay(vsd2)))
sampleDistMatrix2 <- as.matrix(sampleDists2)
rownames(sampleDistMatrix2) <- paste(vsd2$Treatment, vsd2$type, sep="-")
colnames(sampleDistMatrix2) <- NULL

pheatmap(sampleDistMatrix2,
         clustering_distance_rows=sampleDists2,
         clustering_distance_cols=sampleDists2)
```

```{r}
pdf("output/Ind_DE/Plan/Plan_OutlierRemoved_DistanceMatrix.pdf")
  pheatmap(sampleDistMatrix2,
         clustering_distance_rows=sampleDists2,
         clustering_distance_cols=sampleDists2)
dev.off()
```


```{r}
plan.pca2 <- plotPCA(vsd2, intgroup="Treatment", returnData = TRUE)
ggplot(plan.pca2, aes(x = PC1, y = PC2, color = Treatment, label = name)) +
  geom_point() +
  geom_text()
ggsave("output/Ind_DE/Plan/Plan_OutlierRemoved_PCA.pdf")
```

Do we see a difference in number of sig genes in terms of whether they're immune or not ? 

```{r}
sig <- vector(mode = "list", length = length(plan.res))

for (i in 1:length(sig)){
  tmp <- plan.res[[i]][!is.na(plan.res),]
  tmp$Gene <- rownames(tmp)
  sig[[i]] <- tmp$Gene[tmp$padj < 0.1]
  sig[[i]] <- sig[[i]][!is.na(sig[[i]])]
}

run1 <- unique(unlist(sig))
ort1 <- vector(length = length(run1))
ann1 <- vector(length = length(run1))

for(i in 1:length(run1)){
  check <- run1[i] %in% ortho$GeneID
  if (check == FALSE){
  ort1[i] <- "Unassigned"
  ann1[i] <- "Unknown"
} else {
  ort1[i] <- ortho$Orthogroup[ortho$GeneID == run1[i]]
  check2 <- ort1[i] %in% ann.slim$Orthogroup
  if (check2 == FALSE){
    ann1[i] <- "Unknown"
  } else {
  f <- unique(ann.slim$Function[ann.slim$Orthogroup == ort1[i]])
  if (length(f) > 1){
    if(TRUE %in% grepl("Effector|Recognition|Signalling", f)){
      ann1[i] <- "Immune"
    } else {
        if(TRUE %in% grepl("Putative Immune", f)){
          ann1[i] <- "Putative Immune"
        }
      }
  } else {
      f2 <- gsub("Effector", "Immune", 
               gsub("Recognition", "Immune", 
                    gsub("Signalling", "Immune", 
                         gsub("Background", "None Immune", f))))
      ann1[i] <- paste(f2)
      }
  }
  }
}

one <- as.data.frame(cbind(run1, ort1, ann1))
names(one) <- c("Gene", "OrthoGroup", "Function")
one$Run <- paste("One")
```

And again...

```{r}
sig2 <- vector(mode = "list", length = length(plan.res2))

for (i in 1:length(sig2)){
  tmp <- plan.res2[[i]][!is.na(plan.res2),]
  tmp$Gene <- rownames(tmp)
  sig2[[i]] <- tmp$Gene[tmp$padj < 0.1]
  sig2[[i]] <- sig2[[i]][!is.na(sig2[[i]])]
}

run2 <- unique(unlist(sig2))
ort2 <- vector(length = length(run2))
ann2 <- vector(length = length(run2))

for(i in 1:length(run2)){
  check <- run2[i] %in% ortho$GeneID
  if (check == FALSE){
  ort2[i] <- "Unassigned"
  ann2[i] <- "Unknown"
} else {
  ort2[i] <- ortho$Orthogroup[ortho$GeneID == run2[i]]
  check2 <- ort2[i] %in% ann.slim$Orthogroup
  if (check2 == FALSE){
    ann2[i] <- "Unknown"
  } else {
  f <- unique(ann.slim$Function[ann.slim$Orthogroup == ort2[i]])
  if (length(f) > 1){
    if(TRUE %in% grepl("Effector|Recognition|Signalling", f)){
      ann2[i] <- "Immune"
    } else {
        if(TRUE %in% grepl("Putative Immune", f)){
          ann2[i] <- "Putative Immune"
        }
      }
  } else {
      f2 <- gsub("Effector", "Immune", 
               gsub("Recognition", "Immune", 
                    gsub("Signalling", "Immune", 
                         gsub("Background", "None Immune", f))))
      ann2[i] <- paste(f2)
      }
  }
  }
}


two <- as.data.frame(cbind(run2, ort2, ann2))
names(two) <- c("Gene", "OrthoGroup", "Function")
two$Run <- paste("Two")
```


```{r}
one <- as.data.frame(table(one$Function, one$Run))
one$Prop <- one$Freq/sum(one$Freq)
two <- as.data.frame(table(two$Function, two$Run))
two$Prop <- two$Freq/sum(two$Freq)

check.df <- rbind(one,two)

ggplot(check.df, aes(fill=Var1, y=Prop, x=Var2)) + 
    geom_bar(position="dodge", stat="identity")
```
No difference in the proportion of genes... except that there's immune genes appearing in the filtered run.

```{r}
ggplot(check.df, aes(fill=Var1, y=Freq, x=Var2)) + 
    geom_bar(position="dodge", stat="identity")
```
Gonna make the call to go forward without the outliers but .... this may change with Seth's input.

```{r}
plan.dds <- plan.dds2
plan.res <- plan.res2
```


###Results

Save 

```{r}
for (i in 1:length(plan.res)){
  plan.res[[i]]$Contrast <- paste(treats[i])
  plan.res[[i]]$Gene <- rownames(plan.res[[i]])
  rownames(plan.res[[i]]) <- NULL
  plan.res[[i]] <- as.data.frame(plan.res[[i]])
}

plan.res.all <- bind_rows(plan.res[1:3])

id <- read.table("input/Gene_Info/Plan_IDwDesc.tsv", 
                 sep = "\t", quote = "", header = T)

for (i in 1:nrow(plan.res.all)){
  d <- unique(id$Description[id$GeneID == plan.res.all$Gene[i]])
  if(length(d) > 1){
    plan.res.all$Desc[i] <- d[1]
  } else {
    plan.res.all$Desc[i] <- d  
  }
}

write.table(plan.res.all, paste(wd, "Plan_DeSeqResults_All.tsv", sep = ""),
            col.names = T, row.names = F, quote = F, sep = "\t")

plan.res.sig <- subset(plan.res.all, padj < 0.1)

write.table(plan.res.sig, paste(wd, "Plan_DeSeqResults_FDRlt0.1.tsv", sep =""), 
            col.names = T, row.names = F, quote = F, sep = "\t")
```

Genes of interest?

I'm going to take the top 10 most significant and top 10 biggest logFC per treatment. 
Genes that overlap between two (and between the three treatments) will be of particular interest

```{r}
nona <- plan.res.all[!is.na(plan.res.all$padj),]
nona$LogDif <- sub('-', '', nona$log2FoldChange)
tmp <- nona[nona$Contrast == treats[1],]
sig.wound <- head(tmp$Gene[order(tmp$padj)], n = 10)
top.wound <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)
tmp <- nona[nona$Contrast == treats[2],]
sig.pos <- head(tmp$Gene[order(tmp$padj)], n = 10)
top.pos <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)
tmp <- nona[nona$Contrast == treats[3],]
sig.neg <-  head(tmp$Gene[order(tmp$padj)], n = 10)
top.neg <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)

goi <- c(sig.wound, top.wound, sig.pos, top.pos, sig.neg, top.neg)
stat <- c(rep("Top 10 Sig Wound", 10), rep("Top 10 log2FC Wound", 10),
          rep("Top 10 Sig Gram Positive", 10), rep("Top 10 log2FC Gram Positive", 10),
          rep("Top 10 Sig Gram Negative", 10), rep("Top 10 log2FC Gram Negative", 10))

goi.df <- as.data.frame(cbind(goi, stat))


goi.wound <- unique(c(sig.wound, top.wound))
goi.pos <- unique(c(sig.pos, top.pos))
goi.neg <- unique(c(sig.neg, top.neg))
```

###Visualise: Venn Diagrams

```{r}
top.sig <- list("Wound" = sig.wound,
                 "Gram Positive" = sig.pos,
                 "Gram Negative" = sig.neg)

p <- ggvenn(top.sig,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Top 10 Genes per Treatment by adjPvalue")
```

```{r}
dir.create(paste(wd, "Plots/", sep = ""))
ggsave(paste(wd, "Plots/Plan_TopAdjPvalue_Venn.pdf", sep = ""))
```

```{r}
top.lfc <- list("Wound" = top.wound,
                 "Gram Positive" = top.pos,
                 "Gram Negative" = top.neg)

p <- ggvenn(top.lfc,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Top 10 Genes per Treatment by log2FC")
```



```{r}
ggsave(paste(wd, "Plots/Plan_Toplog2FC_Venn.pdf", sep = ""))
```

```{r}
goi <- list("Wound" = goi.wound,
                 "Gram Positive" = goi.pos,
                 "Gram Negative" = goi.neg)

p <- ggvenn(goi,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Genes of Interest Across Treatments")
```


```{r}
ggsave(paste(wd, "Plots/Plan_GOI_Venn.pdf", sep = ""))
```

```{r}
goi.mat <- makeMatrix(goi.df)
goi.mat
```

```{r}
goi.mat$Gene[goi.mat$AllGoi == 1]
```

uncharacterised LOC106790552, tyrosine decarboxylase-like, protein croquemort-like, platelet glycoprotein V-like, uncharacterised LOC106783847, protein FAM133-like, uncharacterised LOC106784133

```{r}
goi.mat$Gene[goi.mat$BacterialGoi == 1]
```

NACHT and WD repeat domain-containing protein 2, uncharacterised LOC106790555

```{r}
goi.mat$Gene[goi.mat$AllSig == 1]
```

uncharacterised LOC106790552, tyrosine decarboxylase-like, protein croquemort-like

```{r}
goi.mat$Gene[goi.mat$AllTop == 1]
```

protein croquemort-like, platelet glycoprotein V-like, uncharacterised LOC106783847, uncharacterised LOC106784133

###Visualise: Expression BoxPlots

```{r}
dir.create(paste(wd, "Plots/ExpressionPlots/", sep = ""))
```

####Significant Wound

```{r}
goi.mat$Gene[goi.mat$SigWound == 1]
genes <- goi.mat$Gene[goi.mat$SigWound == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = plan.dds)
}
```

All Wound: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = plan.dds, 
          filename = "WoundSig")
```


####TopFC Wound

```{r}
goi.mat$Gene[goi.mat$TopWound == 1]
genes <- goi.mat$Gene[goi.mat$TopWound == 1]
```


```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = plan.dds)
}
```

All Wound: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = plan.dds, 
          filename = "Woundlog2FC")
```


####Significant GramPos


```{r}
goi.mat$Gene[goi.mat$SigPos == 1]
genes <- goi.mat$Gene[goi.mat$SigPos == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = plan.dds)
}
```


All Gram Pos Sig: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = plan.dds, 
          filename = "PosSig")
```


####TopFC GramPos


```{r}
goi.mat$Gene[goi.mat$TopPos == 1]
genes <- goi.mat$Gene[goi.mat$TopPos == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = plan.dds)
}
```

All log Gram Pos: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = plan.dds, 
          filename = "Poslog2FC")
```

####Significant GramNeg


```{r}
goi.mat$Gene[goi.mat$SigNeg == 1]
genes <- goi.mat$Gene[goi.mat$SigNeg == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = plan.dds)
}
```

All Gram Neg Sig: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = plan.dds, 
          filename = "NegSig")
```


####TopFC GramPos


```{r}
goi.mat$Gene[goi.mat$TopNeg == 1]
genes <- goi.mat$Gene[goi.mat$TopNeg == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = plan.dds)
}
```

All log Gram Neg: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = plan.dds, 
          filename = "Neglog2FC")
```


###Visualise: PCA

Redo after removing outliers.

```{r}
vsd <- vst(plan.dds, blind = F)

plan.pca <- plotPCA(vsd, intgroup="Treatment", returnData = TRUE)

ggplot(plan.pca, aes(x = PC1, y = PC2, color = Treatment)) +
  geom_point() 
```

```{r}
ggsave(paste(wd, "Plots/Plan_PCA_Simple.pdf", sep = ""))
```

```{r}
ggplot(plan.pca, aes(x = PC1, y = PC2, color = Treatment)) +
  geom_point() +
  geom_label_repel(
    data = plan.pca,
    aes(label = name),
    size = 3
  )
```


```{r}
ggsave(paste(wd, "Plots/Plan_PCA_Labelled.pdf", sep = ""))
```

```{r}
plan.pca$Treat2 <- c(rep("Naive", 2), rep("Treatment", 7))

ggplot(plan.pca, aes(x = PC1, y = PC2, color = Treat2)) +
  geom_point() +
  stat_ellipse()
```

```{r}
ggsave(paste(wd, "Plots/Plan_PCA_TrtVUnTrt.pdf", sep = ""))
```

###Visualise: Smears


```{r}
plot.df <- makeMetaDf(plan.res.all)
plot.df %>%
  filter(Condition == "Wound") %>%
  arrange(padj)
```



```{r}
plot.df$Condition <- factor(plot.df$Condition, levels = c("Wound",
                                                          "Gram Positive",
                                                          "Gram Negative"))

ggplot(plot.df, aes(x = log(baseMean), y = log2FoldChange)) +
  geom_point(aes(colour = Sig)) +
  facet_grid(~Condition) +
  scale_color_manual(values=c("black", "Red"), guide = NULL) +
  theme(legend.position = "bottom")
```



```{r}
ggsave(paste(wd, "Plots/Plan_SmearPlot.pdf", sep = ""))
```


###Visualise: Rainclouds

```{r}
sig.genes <- unique(plot.df$Gene[plot.df$padj < 0.1])
plot.sig <- plot.df[plot.df$Gene %in% sig.genes,]

to.add.naive <- plot.sig[plot.sig$Gene %in% sig.genes &
                               plot.sig$Condition == "Wound",]
to.add.naive$Sig <- "No"
to.add.naive$Condition <- "Naive"
to.add.naive$log2FoldChange <- 0

plot.sig <- rbind(plot.sig, to.add.naive)

plot.sig$Condition <- factor(plot.sig$Condition, levels = c("Naive",
                                                            "Wound", 
                                                            "Gram Positive", 
                                                            "Gram Negative"))

plot.sig$Significance[plot.sig$padj > 0.1] <- "Not Significant"
plot.sig$Significance[plot.sig$padj < 0.1 & plot.sig$padj > 0.05] <- 
  "Significant (FDR < 0.1)"
plot.sig$Significance[plot.sig$padj < 0.05 & plot.sig$padj > 0.01] <-
  "Significant (FDR < 0.05)"
plot.sig$Significance[plot.sig$padj < 0.01 & plot.sig$padj > 0.001] <-
  "Significant (FDR < 0.01)"
plot.sig$Significance[plot.sig$padj < 0.001] <-
  "Significant (FDR < 0.001)"

plot.sig$Significance <- factor(plot.sig$Significance, levels =
                                 c("Not Significant",
                                   "Significant (FDR < 0.1)",
                                   "Significant (FDR < 0.05)",
                                   "Significant (FDR < 0.01)",
                                   "Significant (FDR < 0.001)"))


plot.sig$Function <- factor(plot.sig$Function, levels = c("Immune",
                                                          "Putative Immune", 
                                                          "None Immune", 
                                                          "Unknown"))

ggplot(plot.sig[!plot.sig$Condition == "Naive",], 
       (aes(x = Condition, y = log2FoldChange))) +
  geom_jitter(aes(colour = Function, shape = Significance)) +
  scale_colour_manual(values = c("#2c8cd9", "#7a2cd9", "darkgrey", "plum")) +
  scale_shape_manual(values = c(1, 5, 18, 19, 17)) 
```

```{r}
ggsave(paste(wd, "Plots/Plan_logFCJitter.pdf", sep = ""))
```

```{r}
ggplot(plot.sig, 
       (aes(x = Condition, y = log2FoldChange))) +
  geom_point(aes(colour = Function, shape = Significance)) +
  geom_line( linetype = "dashed", alpha = 0.2, aes(group = Gene, colour = Function)) +
  scale_colour_manual(values = c("#2c8cd9", "#7a2cd9", "darkgrey", "plum")) +
  scale_shape_manual(values = c(1, 5, 18, 19, 17)) 
```

```{r}
ggsave(paste(wd, "Plots/Plan_logFCScatter_GeneLine.pdf", sep = ""))
```

####Visualise: Volcano Plots

```{r}
plot.df$Function <- factor(plot.df$Function, levels = c("Immune",
                                                        "Putative Immune",
                                                        "None Immune",
                                                        "Unknown"))

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.1", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Plan_Simple_FDR0.1_Volcano.pdf", sep = ""))
```


```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.1", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```
```{r}
ggsave(paste(wd, "Plots/Plan_Class_FDR0.1_Volcano.pdf", sep = ""))
```

```{r}
plot.df$Direction2 <- "Not Significant"
plot.df$Direction2[plot.df$padj < 0.05 &
                    plot.df$log2FoldChange > 0 ] <- "Up-Regulated"
plot.df$Direction2[plot.df$padj < 0.05 &
                    plot.df$log2FoldChange < 0 ] <- "Down-Regulated"

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction2), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.05", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```

```{r}
ggsave(paste(wd, "Plots/Plan_Simple_FDR0.05_Volcano.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction2, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.05", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Plan_Class_FDR0.05_Volcano.pdf", sep = ""))
```


```{r}
plot.df$Direction3 <- "Not Significant"
plot.df$Direction3[plot.df$padj < 0.001 &
                    plot.df$log2FoldChange > 0 ] <- "Up-Regulated"
plot.df$Direction3[plot.df$padj < 0.001 &
                    plot.df$log2FoldChange < 0 ] <- "Down-Regulated"

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Plan_Simple_FDR0.001_Volcano.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Plan_Class_FDR0.001_Volcano.pdf", sep = ""))
```

Label top 3 sig genes per treatment

```{r}
sig.wound[1:3]
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "LOC106790552"] <- "Uncharacterised LOC106790552"
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "LOC106787312"] <- "Tyrosine decarboxylase-like"
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "LOC106791417"] <- "Protein croquemort-like"

sig.pos[1:3]
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "LOC106787312"] <- "Tyrosine decarboxylase-like"
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "LOC106793670"] <- "UDP-GUT 1-3-like"
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "LOC106790552"] <- "Uncharacterised LOC106790552"

sig.neg[1:3]
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "LOC106785761"] <- "L(2)EFL-like"
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "LOC106786250"] <- "Protein bicaudal C"
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "LOC106786125"] <- "Inhibin beta chain"
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") +
  geom_label_repel(
    data = plot.df[!is.na(plot.df$Label),],
    size = 2,
    aes(label = Label),
    point.padding = unit(0.3, "lines"),
    box.padding = 1.5
  ) 

```


```{r}
ggsave(paste(wd, "Plots/Plan_Simple_FDR0.001_Volcano_top3Label.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.01", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") +
  geom_label_repel(
    data = plot.df[!is.na(plot.df$Label),],
    size = 2,
    aes(label = Label),
    point.padding = unit(0.3, "lines"),
    box.padding = 1.5
  ) 

```

```{r}
ggsave(paste(wd, "Plots/Plan_Class_FDR0.001_Volcano_top3Label.pdf", sep = ""))
```

##SessionInfo

```{r}
writeLines(capture.output(sessionInfo()),
           "SessionLogs/Proj_Pt2_SesInfo.txt")
sessionInfo()
```










