---
title: "Comp_Hym_Proj_Jul22_5"
output: html_document
date: "2022-07-20"
---


20th July 2022

OrthoGroup level differential expression analysis.

Runs with outliers (two combinations) removed.

```{r}
set.seed(42)
```

##Libraries

```{r}
library("DESeq2")
library("tidyverse")
library("ggrepel")
library("tximport")
library("pheatmap")
library("reshape")
library("ggvenn")
```

##Resources

Following recommended workflow: http://master.bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html#preparing-quantification-input-to-deseq2

DeSeq2: https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html

Trim Galore:  https://github.com/FelixKrueger/TrimGalore

Kallisto :    https://pachterlab.github.io/kallisto/about

Gene level instructions from tximport: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4712774/

OrthoGroup information / Annotation information

```{r}
ortho <- read.table("input/Ortho_Files/OrthoFinder_Combined_Parsed_Jul22.tsv",
                    sep = "\t", header = T)

ann <- read.table("input/Gene_Info/Amel_ImmAnnotations_Orthogroup_Jun22.tsv",
                  header = T, sep = "\t")
ann.slim <- ann[,c(2:3)]
```

Fix Kr-h1 issue (distinction not need for this run)

```{r}
ortho$GeneID[grep("Kr-h1", ortho$GeneID)] <- "Kr-h1"
```

```{r}
ortgps <- as.data.frame(unique(ortho$Orthogroup))

for(i in 1:nrow(ortgps)){
  species <- ortho$Species[ortho$Orthogroup == ortgps[i,1]]
  ortgps[i,2] <- ifelse("Amel" %in% species, 1, 0)
  ortgps[i,3] <- ifelse("Bter" %in% species, 1, 0)
  ortgps[i,4] <- ifelse("Caus" %in% species, 1, 0)
  ortgps[i,5] <- ifelse("Pcan" %in% species, 1, 0)
}

names(ortgps) <- c("OrthoGroup", "Amel", "Bter",
                   "Caus", "Plan")
```

Make tx object with orthogroup IDs in place of GeneIDs

```{r}
runs <- c("Amel", "Bter", "Caus", "Plan")

tx.list <- vector(mode = "list", length = length(runs))

for (i in 1:length(runs)){
  tx.list[[i]] <- read.table(paste("input/Gene_Info/", runs[i],
                                   "_TXGene.tsv", sep = ""),
                             header = T, sep = "\t")
  for (j in 1:nrow(tx.list[[i]])){
    check <- tx.list[[i]][j,2] %in% ortho$GeneID
    if (check == FALSE){
      tx.list[[i]]$ORTHOID[j] <- paste("Unassigned", runs[i], sep = "_")
    } else {
      tx.list[[i]]$ORTHOID[j] <- unique(ortho$Orthogroup
                                        [ortho$GeneID == tx.list[[i]][j,2]])
    }
  }
  tx.list[[i]] <- tx.list[[i]][,c(1,3)]
}

ort.tx <- bind_rows(tx.list)
```

##Functions

Function that takes in raw abundance.tsv files and keeps only those transcripts that are members of orthogroups that have representatives of all input species. These are then collected into a folder reader to be read into tximport.

```{r}
filterCount <- function(specieslist, outname){
    #object read to read in orthogroups that have members in all given species
    ort.grp <- vector(mode = "list", length = length(specieslist))
    #iterate through empty list and fill each with list of orthogroups 
    #that species has been assigend to
    for (i in 1:length(ort.grp)){
      ort.grp[[i]] <- ortgps$OrthoGroup[ortgps[,paste(specieslist[i])] == 1]
    }
    #combine list of orthogroups into one vector
    ort.grp <- unlist(ort.grp)
    #put into table format with orthogroup as dimname and frequency as value
    #ie the number of species assigned to that orthogroup
    x <- table(ort.grp)
    #how many orthogroups are present that have all the species listed as input included
    tot <- length(names(x[x == length(specieslist)]))
    #print to screen
    print(paste("There are", tot, "orthogroups shared between these species", sep = " "))
    #record these target orthogroups
    tar.ort <- names(x[x==length(specieslist)])
    #collect a list of transcripts associated with these orthogroups
    tx.out <- ort.tx$TXNAME[ort.tx$ORTHOID %in% tar.ort]
    #prepare to produce output
    dir.create(paste("output/Ortho_DE/", outname, sep = ""))
    dir.create(paste("output/Ortho_DE/", outname, "/Filtered_Counts/", sep = ""))
    #for each species, read in each samples abundance file and keep only rows
    #associated with transcripts that correspond with target orthogroups
    for (s in 1:length(specieslist)){
      dir <- paste("input/Counts_Kal/", specieslist[s], sep = "")
      files <- list.files(dir)
      for (f in 1:length(files)){
        entrer <- read.table(paste(dir, files[f], sep = "/"),
                             header = T, sep = "\t")
        sortir <- entrer[entrer$target_id %in% tx.out,]
        splitf <- strsplit(files[f], ".", fixed =T)
        fname <- paste(splitf[[1]][1], ".orthoFiltered_", 
                       splitf[[1]][2], ".", splitf[[1]][3],
                        sep = "")
        write.table(sortir,
                    paste("output/Ortho_DE/", outname, "/Filtered_Counts/", 
                          fname, sep =""),
                    col.names = T, row.names = F, quote = F, sep ="\t")
      }
    }
}
```

A function that reads all filtered count files in a given orthogroup work directory and feed them into tximport in order to produce a list of txobjects (one per input species). Coldata object is produced from file names.

```{r}
ortCount.outRem <- function(ortrun, txgeneobjlist, outlier_s){
  #store the directory where files needed to be extracted from
  dir <- paste("output/Ortho_DE/", ortrun, "/Filtered_Counts", sep = "")
  #list the files within this directory (count files to be filtered)
  files <- list.files(dir)
  #extract the samples from the file names
  samples <- sapply(strsplit(files, ".", fixed = T), "[", 1)
  #name the files with sample IDS
  names(files) <- samples
  #remove outliers
  files <- files[!names(files) %in% outlier_s]
  samples <- samples[!samples %in% outlier_s]
  #prepare to make coldata object using codes in sample names for species and treatment
  treat <- vector(length = length(samples))
  species <- vector(length = length(samples))
  #store treatments per sample
  for (i in 1:length(samples)){
    if(grepl("_N_", samples[i]) == T){
      treat[i] <- "Naive"
    } else {
      if(grepl("_P_", samples[i]) == T){
        treat[i] <- "Wound"
      } else {
        if(grepl("_SM_", samples[i]) == T){
          treat[i] <- "GramPos"
        } else {
          if(grepl("_SL_", samples[i]) == T){
            treat[i] <- "GramNeg"
          }
        }
      }    
    }
  }
  #store species per sample
  for (i in 1:length(samples)){
    if(grepl("AM_", samples[i]) == T){
      species[i] <- "Amel"
    } else {
      if(grepl("BT_", samples[i]) == T){
        species[i] <- "Bter"
      } else {
        if(grepl("CA_", samples[i]) == T){
          species[i] <- "Caus"
        } else {
          if(grepl("PL_", samples[i]) == T){
            species[i] <- "Plan"
          }
        }
      }    
    }
  }
  #combine these vectors into a dataframe which will be coldata object for subsequent
  #analyses
  cd <- as.data.frame(cbind(samples, treat, species))
  names(cd) <- c("Samples", "Treatment", "Species")
  #write up coldata
  write.table(cd, 
              paste("output/Ortho_DE/", ortrun, "/", ortrun, "_SampleData.tsv", sep = ""),
              col.names = T, row.names = F, sep = "\t", quote = F)
  cnts <- file.path(dir, files)
  names(cnts) <- samples
  #each species will have its own txobject that will be combined into one for DE
  sets <- unique(species)
  tx.set <- vector(mode = "list", length = length(sets))
  #run through each species "set", reading in only those abundance files
  for (s in 1:length(sets)){
    print(paste(sets[s], "..." ))
    samp.set <- cd$Samples[cd$Species == paste(sets[s])]
    #print the samples used to make sure correct outliers have been removed
    print(samp.set)
    cd.set <- cd[cd$Species == paste(sets[s]),]
    txgene.set <- txgeneobjlist[[s]]
    cnts.set <- cnts[names(cnts) %in% samp.set]
    tx.set[[s]] <- tximport(cnts.set, type = "kallisto",
                            tx2gene = txgene.set, ignoreAfterBar = T)
  }
  return(tx.set)
}
```

Produce DDS object from count table object (includes making coldata object from column names, prefiltering and setting reference condition to Naive)

```{r}
getDDS <- function(ortrun, txobject){
  cd <- read.table(paste("output/Ortho_DE/", ortrun, "/",
                         ortrun, "_SampleData.tsv", sep = ""), header = T, sep = "\t")
  ddsTx <- DESeqDataSetFromTximport(txobject, colData = cd, design = ~Treatment)
  keep <- rowSums(counts(ddsTx)) >= 10
  ddsTx2 <- ddsTx[keep,]
  ddsTx2$Treatment <- relevel(ddsTx2$Treatment, ref = "Naive")
  ddsTx2 <- DESeq(ddsTx2)
  return(ddsTx2)
}
```

Function to plot expression boxplot of a certain orthogroup separating by species. Includes an all species variable.

```{r}
visiOrt.messy <- function(orthogroup, ortname, dds){
  tmp <- counts(dds, normalized = T)
  samples <- dds$Samples
  counts <- tmp[rownames(tmp) == paste(orthogroup),]
  plot.df <- as.data.frame(cbind(samples, counts))
  names(plot.df) <- c("Samples", "Norm_Counts")
  plot.df$Species[grepl("AM_", plot.df$Samples)] <- "A mellifera"
  plot.df$Species[grepl("BT_", plot.df$Samples)] <- "B terrestris"
  plot.df$Species[grepl("CA_", plot.df$Samples)] <- "C australensis"
  plot.df$Species[grepl("PL_", plot.df$Samples)] <- "P lanio"
  plot.df$Treatment[grepl("_N_", plot.df$Samples)] <- "Naive"
  plot.df$Treatment[grepl("_P_", plot.df$Samples)] <- "Wound"
  plot.df$Treatment[grepl("_SM_", plot.df$Samples)] <- "Gram Negative"
  plot.df$Treatment[grepl("_SL_", plot.df$Samples)] <- "Gram Positive"
  plot.df$Treatment <- factor(plot.df$Treatment, levels = c("Naive",
                                                        "Wound", 
                                                        "Gram Positive", 
                                                        "Gram Negative"))
  plot.df$Norm_Counts <- as.numeric(plot.df$Norm_Counts)
  to.add <-plot.df
  to.add$Species <- "All Species"
  plot.df2 <- rbind(plot.df,to.add)
  plot.df2$Species <- factor(plot.df2$Species, levels = c("All Species",
                                                "A mellifera",
                                                "B terrestris",
                                                "C australensis",
                                                "P lanio"))
  p <- ggplot(plot.df2, aes(x = Treatment, y = log(Norm_Counts), fill = Species)) +
      geom_boxplot() +
      scale_fill_manual(values = c("grey", "#ffa758",  "#58b0ff", "#58ffa7", "#ff58b0")) +
      xlab("Treatment") +
      ylab("log(DESeq2 Normalised Counts)") +
      labs(title = (paste(ortname, " expression across treatments", sep = ""))) +
      guides(alpha = FALSE, colour = FALSE)
    ggsave(paste(wd, orthogroup, "_ExpressionBoxplot.messy.pdf", sep = ""))
    p
}
```

Function to plot expression boxplot of a certain orthogroup separating by treatment. Species are plotted as different shaped geom_points.

```{r}
visiOrt.clean <- function(orthogroup, ortname, dds){
  tmp <- counts(dds, normalized = T)
  samples <- dds$Samples
  counts <- tmp[rownames(tmp) == paste(orthogroup),]
  plot.df <- as.data.frame(cbind(samples, counts))
  names(plot.df) <- c("Samples", "Norm_Counts")
  plot.df$Species[grepl("AM_", plot.df$Samples)] <- "A mellifera"
  plot.df$Species[grepl("BT_", plot.df$Samples)] <- "B terrestris"
  plot.df$Species[grepl("CA_", plot.df$Samples)] <- "C australensis"
  plot.df$Species[grepl("PL_", plot.df$Samples)] <- "P lanio"
  plot.df$Treatment[grepl("_N_", plot.df$Samples)] <- "Naive"
  plot.df$Treatment[grepl("_P_", plot.df$Samples)] <- "Wound"
  plot.df$Treatment[grepl("_SM_", plot.df$Samples)] <- "Gram Negative"
  plot.df$Treatment[grepl("_SL_", plot.df$Samples)] <- "Gram Positive"
  plot.df$Treatment <- factor(plot.df$Treatment, levels = c("Naive",
                                                        "Wound", 
                                                        "Gram Positive", 
                                                        "Gram Negative"))
  plot.df$Norm_Counts <- as.numeric(plot.df$Norm_Counts)
  plot.df$Species <- factor(plot.df$Species, levels = c(
                                                "A mellifera",
                                                "B terrestris",
                                                "C australensis",
                                                "P lanio"))
  p <- ggplot(plot.df, aes(x = Treatment, y = log(Norm_Counts), fill = Treatment)) +
      geom_boxplot(alpha = .8, outlier.colour = "red") +
      geom_point(aes(shape = Species), position = position_dodge(width = .2),
                 size = 2) +
      scale_fill_manual(values = c("#ffa758",  "#58b0ff", "#58ffa7", "#ff58b0")) +
      scale_shape_manual(values = c(1, 0, 2, 4))+
      xlab("Treatment") +
      ylab("log(DESeq2 Normalised Counts)") +
      labs(title = (paste(ortname, " expression across treatments", sep = ""))) +
      guides(alpha = FALSE, colour = FALSE)
    ggsave(paste(wd, orthogroup, "_ExpressionBoxplot.clean.pdf", sep = ""))
    p
}
```


Produce a dataframe with orthogroup and gene class information ready for data visualisation

```{r}
makeMetaDf <- function(res.all.object){
  x <- res.all.object[!is.na(res.all.object$padj),]
  x$Condition[x$Contrast==paste(treats[1])] <- "Wound"
  x$Condition[x$Contrast==paste(treats[2])] <- "Gram Positive"
  x$Condition[x$Contrast==paste(treats[3])] <- "Gram Negative"
  for(i in 1:nrow(x)){
  check <- x$Gene[i] %in% ortho$GeneID
    if (check == FALSE){
      x$OrthoGroup[i] <- "Unassigned"
    } else {
    x$OrthoGroup[i] <- ortho$Orthogroup[ortho$GeneID ==
                                                       x$Gene[i]]
    }
  }
  for(i in 1:nrow(x)){
  if (x$OrthoGroup[i] == "Unassigned") {
    x$Function[i] <- "Unknown"} else {
      check2 <- x$OrthoGroup[i] %in% ann.slim$Orthogroup
      if (check2 == FALSE){
        x$Function[i] <- "Unknown"
      } else {
    f <- unique(ann.slim$Function[ann.slim$Orthogroup == x$OrthoGroup[i]])
    if (length(f) > 1){
      if(TRUE %in% grepl("Effector|Recognition|Signalling", f)){
        x$Function[i] <- "Immune"
    } else {
      if(TRUE %in% grepl("Putative Immune", f)){
        x$Function[i] <- "Putative Immune"
        } 
      }
    } else {
    f2 <- gsub("Effector", "Immune", 
               gsub("Recognition", "Immune", 
                    gsub("Signalling", "Immune", 
                         gsub("Background", "None Immune", f))))
    x$Function[i] <- paste(f2)
        }
      }
    }
  }
  x$Sig <- "No"
  x$Sig[x$padj < 0.1] <- "Yes"
  x$Condition <- factor(x$Condition, levels = c("Wound", 
                                                "Gram Negative", 
                                                "Gram Positive"))
  x$Neglog10adjPvalue <- -log10(x$padj)
  x$Direction <- "Not Significant"
  x$Direction[x$Sig == "Yes" &
                    x$log2FoldChange > 0 ] <- "Up-Regulated"
  x$Direction[x$Sig == "Yes" &
                    x$log2FoldChange < 0 ] <- "Down-Regulated"
  return(x)
}
```

Produce a 1/0 binary matrix for use in comparing overlap in genes of interest

```{r}
makeMatrix <- function(goidf){
  mat <- as.data.frame(unique(goidf$goi))
  names(mat) <- "Gene"
  mat$SigWound <- ifelse(mat$Gene %in% sig.wound, 1, 0)
  mat$SigPos <- ifelse(mat$Gene %in% sig.pos, 1, 0)
  mat$SigNeg <- ifelse(mat$Gene %in% sig.neg, 1, 0)
  mat$TopWound <- ifelse(mat$Gene %in% top.wound, 1, 0)
  mat$TopPos <- ifelse(mat$Gene %in% top.pos, 1, 0)
  mat$TopNeg <- ifelse(mat$Gene %in% top.neg, 1, 0)
  mat$AllSig <- ifelse(mat$SigWound == 1 &
                      mat$SigPos == 1 &
                      mat$SigNeg == 1, 1, 0)
  mat$AllTop <- ifelse(mat$TopWound == 1 &
                      mat$TopPos == 1 &
                      mat$TopNeg == 1, 1, 0)
  mat$BacterialSig <- ifelse(mat$SigWound == 0 &
                      mat$SigPos == 1 &
                      mat$SigNeg == 1, 1, 0)
  mat$BacterialTop <- ifelse(mat$TopWound == 0 &
                      mat$TopPos == 1 &
                      mat$TopNeg == 1, 1, 0)
  mat$GoiWound <- ifelse(mat$TopWound == 1 | mat$SigWound == 1, 1, 0)
  mat$GoiPos <- ifelse(mat$TopPos == 1 | mat$SigPos == 1, 1, 0)
  mat$GoiNeg <- ifelse(mat$TopNeg == 1 | mat$SigNeg == 1, 1, 0)
  mat$AllGoi <- ifelse(mat$GoiWound == 1 &
                      mat$GoiPos == 1 &
                        mat$GoiNeg == 1, 1, 0)
  mat$BacterialGoi <- ifelse(mat$GoiPos == 1 &
                        mat$GoiNeg == 1  &
                          mat$GoiWound == 0, 1, 0)
  return(mat)
}
```

Produce and save a boxplot with expression levels across samples with gene input

```{r}
visiGene <- function(gene, genename, dds){
  p <- plotCounts(dds, gene = paste(gene), intgroup = "Treatment", returnData = T)
  p$Treatment2 <- as.character(p$Treatment)
  p$Treatment2[p$Treatment == "GramPos"] <- "Gram Positive"
  p$Treatment2[p$Treatment == "GramNeg"] <- "Gram Negative"  
  p$Treatment2 <- factor(p$Treatment2, levels = c("Naive",
                                              "Wound", 
                                              "Gram Positive", 
                                              "Gram Negative"))  
  p2 <- ggplot(p, aes(x = Treatment2, y = count)) +
    geom_boxplot(aes(colour = Treatment2, alpha = 0.5)) +
    geom_point(position = position_jitter(w=0.1, h = 0),
               aes(colour = Treatment2), size = 3) +
    scale_colour_manual(values = c("#58ffa7", "#ffa758", "#ff58b0", "#58b0ff")) +
    xlab("Treatment") +
    labs(title = (paste(genename, " expression across treatments", sep = ""))) +
    guides(alpha = FALSE, colour = guide_legend("Treatment"))
  ggsave(paste(wd, "Plots/ExpressionPlots/", gene, "_ExpBoxPlot.pdf", sep = ""))
  p2
}
```

Plot all expression boxplots from a group of genes 

```{r}
visiFacet <- function(genes, genenames, dds, filename){
  p <- vector(mode = "list", length = length(genes))
  for (i in 1:length(p)){
    p[[i]] <- plotCounts(dds, gene = genes[i], 
                         intgroup = "Treatment", returnData = T)
    p[[i]]$Gene <- paste(genes[i])
    p[[i]]$Label <- paste(genenames[i])
  }
  p2 <- bind_rows(p)
  p2$Treatment2 <- as.character(p2$Treatment)
  p2$Treatment2[p2$Treatment == "GramPos"] <- "Gram Positive"
  p2$Treatment2[p2$Treatment == "GramNeg"] <- "Gram Negative"  
  p2$Treatment2 <- factor(p2$Treatment2, levels = c("Naive",
                                              "Wound", 
                                            "Gram Positive", 
                                              "Gram Negative"))  
  p3 <- ggplot(p2, aes(x = Treatment2, y = count)) +
    geom_boxplot(aes(colour = Treatment2, alpha = 0.5)) +
    geom_point(position = position_jitter(w=0.1, h = 0),
               aes(colour = Treatment2), size = 3) +
    scale_colour_manual(values = c("#58ffa7", "#ffa758", "#ff58b0", "#58b0ff")) +
    xlab("Treatment") +
    guides(alpha = FALSE, colour = guide_legend("Treatment")) +
    facet_wrap(~Label, nrow = 5, scale = "free_y")
  ggsave(paste(wd, "Plots/ExpressionPlots/", filename, "_Facet_ExpBoxPlot.pdf", sep = ""))
  p3
}
```



##OrthoLevel DE: All Species (Aculeata) sans Bter outliers

###Preparing TX Object

Looking at orthogroups present in all four species.

First, produce files that have non-orthogroup associated transcripts removed and save them in output/Ortho_DE/Aculeata/ ready to be ready to be fed into tximport.

```{r}
runs <- c("Amel", "Bter", "Caus", "Plan")
filterCount(runs, "Aculeata_sB")
```

There's three ways to run this: with all species, with only Bter outliers removed, and with all putative outliers removed.

This time, I'll only remove Bter outliers.

```{r}
wd <- "output/Ortho_DE/Aculeata_sB/"
outliers <- c("BT_A_N_001", "BT_A_P_030", "BT_A_SL_003", "BT_A_SM_032")
```


And now to make a list of txobjects (one per species) from these filtered count tables.

```{r}
ort.acu.sb.txs <- ortCount.outRem("Aculeata_sB", tx.list, outliers)

#make combined abundance matrix
abundance <- cbind(ort.acu.sb.txs[[1]]$abundance,
                   ort.acu.sb.txs[[2]]$abundance,
                   ort.acu.sb.txs[[3]]$abundance,
                   ort.acu.sb.txs[[4]]$abundance)
nrow(abundance)
```

```{r}
#make combined counts matrix
counts <- cbind(ort.acu.sb.txs[[1]]$counts,
                   ort.acu.sb.txs[[2]]$counts,
                   ort.acu.sb.txs[[3]]$counts,
                   ort.acu.sb.txs[[4]]$counts)
nrow(counts)
```

```{r}
#make combined lengths matrix
length <- cbind(ort.acu.sb.txs[[1]]$length,
                   ort.acu.sb.txs[[2]]$length,
                   ort.acu.sb.txs[[3]]$length,
                   ort.acu.sb.txs[[4]]$length)
nrow(length)
```

```{r}
ort.acu.sb.tx <- list(abundance = abundance,
                  counts = counts, 
                  length = length)

ort.acu.sb.tx$countsFromAbundance <- "no"

for (i in 1:length(ort.acu.sb.tx)){
  print(head(ort.acu.sb.tx[[i]], n = 1))
}
```

```{r}
remove(ort.acu.sb.txs)
```


###DE Analysis

```{r}
ort.acu.sb.dds <- getDDS("Aculeata_sB", ort.acu.sb.tx)
```

```{r}
ort.acu.sb.res <- vector(mode = "list", length = 3)
treats <- resultsNames(ort.acu.sb.dds)[c(4,3,2)]

for(i in 1:length(ort.acu.sb.res)){
  ort.acu.sb.res[[i]] <- results(ort.acu.sb.dds, name = paste(treats[i]))
  print(treats[i])
  summary(ort.acu.sb.res[[i]])
}
```

###Results

Save 

```{r}
for (i in 1:length(ort.acu.sb.res)){
  ort.acu.sb.res[[i]]$Contrast <- paste(treats[i])
  ort.acu.sb.res[[i]]$OrthoGroup <- rownames(ort.acu.sb.res[[i]])
  rownames(ort.acu.sb.res[[i]]) <- NULL
  ort.acu.sb.res[[i]] <- as.data.frame(ort.acu.sb.res[[i]])
}

ort.acu.sb.res.all <- bind_rows(ort.acu.sb.res[1:3])

write.table(ort.acu.sb.res.all, paste(wd, "Aculeata_sB_DeSeqResults_All.tsv", sep = ""),
            col.names = T, row.names = F, quote = F, sep = "\t")

ort.acu.sb.res.sig <- subset(ort.acu.sb.res.all, padj < 0.1)

write.table(ort.acu.sb.res.sig, paste(wd, "Aculeata_sB_DeSeqResults_FDRlt0.1.tsv", sep =""), 
            col.names = T, row.names = F, quote = F, sep = "\t")
```

###ExpBoxPlots

What is signficant ? 

```{r}
ort.acu.sb.res.sig
```


```{r}
sig.ort <- ort.acu.sb.res.sig$OrthoGroup
ann[ann$Orthogroup %in% sig.ort,]
```

OG0000415: Trypsin

Putative immune orthogroup, sig (FDR < 0.05) in Gram Pos treatment

```{r}
visiOrt.messy("OG0000415", "Trypsin", ort.acu.sb.dds)
```


```{r}
visiOrt.clean("OG0000415", "Trypsin", ort.acu.sb.dds)
```

OG0003982: leukocyte elastase inhibitor (serpin)

Canonical immune signaling orthogroup sig (FDR < 0.1) in Gram Pos treatment
Sig (FDR < 0.01) in Gram Neg


```{r}
visiOrt.messy("OG0003982", "Leukocyte elastase inhibitor", ort.acu.sb.dds)
```

```{r}
visiOrt.clean("OG0003982", "Leukocyte elastase inhibitor", ort.acu.sb.dds)
```

OG0004924: ninjurin-1

Background gene sig (FDR < 0.05) in Gram Pos treatment

```{r}
visiOrt.messy("OG0004924", "Ninjurin-1", ort.acu.sb.dds)
```


```{r}
visiOrt.clean("OG0004924", "Ninjurin-1", ort.acu.sb.dds)
```
OG0005892: uncharacterised OG0005892

Background gene sig (FDR < 0.05) in Gram Pos treatment

```{r}
visiOrt.messy("OG0005892", "Uncharcterised OG0005892", ort.acu.sb.dds)
```


```{r}
visiOrt.clean("OG0005892", "Uncharacterised OG0005892", ort.acu.sb.dds)
```

OG0004924: ninjurin-1

Background gene sig (FDR < 0.05) in Gram Pos treatment

```{r}
visiOrt.messy("OG0004924", "Ninjurin-1", ort.acu.sb.dds)
```


```{r}
visiOrt.clean("OG0004924", "Ninjurin-1", ort.acu.sb.dds)
```



Make some space

```{r}
remove(ort.acu.sb.dds, ort.acu.sb.res, 
       ort.acu.sb.res.all, ort.acu.sb.res.sig, 
       ort.acu.sb.tx)
```

##OrthoLevel DE: All Species (Aculeata) sans all outliers

###Preparing TX Object

Looking at orthogroups present in all four species.

First, produce files that have non-orthogroup associated transcripts removed and save them in output/Ortho_DE/Aculeata/ ready to be ready to be fed into tximport.

```{r}
runs <- c("Amel", "Bter", "Caus", "Plan")
filterCount(runs, "Aculeata_sA")
```

There's three ways to run this: with all species, with only Bter outliers removed, and with all putative outliers removed.

This time, I'll only remove Bter outliers.

```{r}
wd <- "output/Ortho_DE/Aculeata_sA/"
outliers <- c("BT_A_N_001", "BT_A_P_030", "BT_A_SL_003", "BT_A_SM_032",
              "CA_N_012", "CA_P_055", "CA_SL_056",
              "PL_N_1908", "PL_P_1912", "PL_SM_1936")
```


And now to make a list of txobjects (one per species) from these filtered count tables.

```{r}
ort.acu.sa.txs <- ortCount.outRem("Aculeata_sA", tx.list, outliers)

#make combined abundance matrix
abundance <- cbind(ort.acu.sa.txs[[1]]$abundance,
                   ort.acu.sa.txs[[2]]$abundance,
                   ort.acu.sa.txs[[3]]$abundance,
                   ort.acu.sa.txs[[4]]$abundance)
nrow(abundance)
```

```{r}
#make combined counts matrix
counts <- cbind(ort.acu.sa.txs[[1]]$counts,
                   ort.acu.sa.txs[[2]]$counts,
                   ort.acu.sa.txs[[3]]$counts,
                   ort.acu.sa.txs[[4]]$counts)
nrow(counts)
```

```{r}
#make combined lengths matrix
length <- cbind(ort.acu.sa.txs[[1]]$length,
                   ort.acu.sa.txs[[2]]$length,
                   ort.acu.sa.txs[[3]]$length,
                   ort.acu.sa.txs[[4]]$length)
nrow(length)
```

```{r}
ort.acu.sa.tx <- list(abundance = abundance,
                  counts = counts, 
                  length = length)

ort.acu.sa.tx$countsFromAbundance <- "no"

for (i in 1:length(ort.acu.sa.tx)){
  print(head(ort.acu.sa.tx[[i]], n = 1))
}
```

```{r}
remove(ort.acu.sa.txs)
```


###DE Analysis

```{r}
ort.acu.sa.dds <- getDDS("Aculeata_sA", ort.acu.sa.tx)
```

```{r}
ort.acu.sa.res <- vector(mode = "list", length = 3)
treats <- resultsNames(ort.acu.sa.dds)[c(4,3,2)]

for(i in 1:length(ort.acu.sa.res)){
  ort.acu.sa.res[[i]] <- results(ort.acu.sa.dds, name = paste(treats[i]))
  print(treats[i])
  summary(ort.acu.sa.res[[i]])
}
```

###Results

Save 

```{r}
for (i in 1:length(ort.acu.sa.res)){
  ort.acu.sa.res[[i]]$Contrast <- paste(treats[i])
  ort.acu.sa.res[[i]]$OrthoGroup <- rownames(ort.acu.sa.res[[i]])
  rownames(ort.acu.sa.res[[i]]) <- NULL
  ort.acu.sa.res[[i]] <- as.data.frame(ort.acu.sa.res[[i]])
}

ort.acu.sa.res.all <- bind_rows(ort.acu.sa.res[1:3])

write.table(ort.acu.sa.res.all, paste(wd, "Aculeata_sA_DeSeqResults_All.tsv", sep = ""),
            col.names = T, row.names = F, quote = F, sep = "\t")

ort.acu.sa.res.sig <- subset(ort.acu.sa.res.all, padj < 0.1)

write.table(ort.acu.sa.res.sig, paste(wd,
                                      "Aculeata_sA_DeSeqResults_FDRlt0.1.tsv", sep =""), 
            col.names = T, row.names = F, quote = F, sep = "\t")
```

###ExpBoxPlots

What is signficant ? 

```{r}
ort.acu.sa.res.sig
```


```{r}
sig.ort <- ort.acu.sa.res.sig$OrthoGroup
ann[ann$Orthogroup %in% sig.ort,]
```

OG0000699: Cactus

Canon signalling gene, sig (FDR < 0.05) in Gram Pos, < 0.1 Gram Neg treatment

```{r}
visiOrt.messy("OG0000699", "Cactus", ort.acu.sa.dds)
```


```{r}
visiOrt.clean("OG0000699", "Cactus", ort.acu.sa.dds)
```
OG0002392: Dopa decarboxylasae

Background gene, sig (FDR < 0.1) in Gram Pos

```{r}
visiOrt.messy("OG0002392", "Dopa decarboxylasae", ort.acu.sa.dds)
```


```{r}
visiOrt.clean("OG0002392", "Dopa decarboxylasae", ort.acu.sa.dds)
```

OG0003982: leukocyte elastase inhibitor (serpin)

Canonical immune signaling orthogroup sig (FDR < 0.01) in Gram Neg treatment and < 0.05 Gram Pos


```{r}
visiOrt.messy("OG0003982", "Leukocyte elastase inhibitor", ort.acu.sa.dds)
```


```{r}
visiOrt.clean("OG0003982", "Leukocyte elastase inhibitor", ort.acu.sa.dds)
```
OG0004924: ninjurin-1

Background gene sig (FDR < 0.05) in Gram Pos treatment

```{r}
visiOrt.messy("OG0004924", "Ninjurin-1", ort.acu.sa.dds)
```


```{r}
visiOrt.clean("OG0004924", "Ninjurin-1", ort.acu.sa.dds)
```


OG0005280: Art3

Background gene, sig (FDR < 0.1) in Gram Pos

```{r}
visiOrt.messy("OG0005280", "Art3", ort.acu.sa.dds)
```


```{r}
visiOrt.clean("OG0005280", "Art3", ort.acu.sa.dds)
```



OG0000399: Toll

Canonical immune signaling orthogroup sig (FDR < 0.05) in Gram Neg treatment


```{r}
visiOrt.messy("OG0000399", "Toll", ort.acu.sa.dds)
```


```{r}
visiOrt.clean("OG0000399", "Toll", ort.acu.sa.dds)
```


Make some space

```{r}
remove(ort.acu.sa.dds, ort.acu.sa.res, 
       ort.acu.sa.res.all, ort.acu.sa.res.sig, 
       ort.acu.sa.tx)
```




##OrthoLevel DE: All Bee (Anthophila) sans Bter outliers

###Preparing TX Object

Looking at orthogroups present in all four species.

First, produce files that have non-orthogroup associated transcripts removed and save them in output/Ortho_DE/Aculeata/ ready to be ready to be fed into tximport.

```{r}
runs <- c("Amel", "Bter", "Caus")
filterCount(runs, "Anthophila_sB")
```

There's three ways to run this: with all species, with only Bter outliers removed, and with all putative outliers removed.

This time, I'll only remove Bter outliers.

```{r}
wd <- "output/Ortho_DE/Anthophila_sB/"
outliers <- c("BT_A_N_001", "BT_A_P_030", "BT_A_SL_003", "BT_A_SM_032")
```


And now to make a list of txobjects (one per species) from these filtered count tables.

```{r}
ort.ant.sb.txs <- ortCount.outRem("Anthophila_sB", tx.list, outliers)

#make combined abundance matrix
abundance <- cbind(ort.ant.sb.txs[[1]]$abundance,
                   ort.ant.sb.txs[[2]]$abundance,
                   ort.ant.sb.txs[[3]]$abundance)
nrow(abundance)
```

```{r}
#make combined counts matrix
counts <- cbind(ort.ant.sb.txs[[1]]$counts,
                   ort.ant.sb.txs[[2]]$counts,
                   ort.ant.sb.txs[[3]]$counts)
nrow(counts)
```

```{r}
#make combined lengths matrix
length <- cbind(ort.ant.sb.txs[[1]]$length,
                   ort.ant.sb.txs[[2]]$length,
                   ort.ant.sb.txs[[3]]$length)
nrow(length)
```

```{r}
ort.ant.sb.tx <- list(abundance = abundance,
                  counts = counts, 
                  length = length)

ort.ant.sb.tx$countsFromAbundance <- "no"

for (i in 1:length(ort.ant.sb.tx)){
  print(head(ort.ant.sb.tx[[i]], n = 1))
}
```

```{r}
remove(ort.ant.sb.txs)
```


###DE Analysis

```{r}
ort.ant.sb.dds <- getDDS("Anthophila_sB", ort.ant.sb.tx)
```

```{r}
ort.ant.sb.res <- vector(mode = "list", length = 3)
treats <- resultsNames(ort.ant.sb.dds)[c(4,3,2)]

for(i in 1:length(ort.ant.sb.res)){
  ort.ant.sb.res[[i]] <- results(ort.ant.sb.dds, name = paste(treats[i]))
  print(treats[i])
  summary(ort.ant.sb.res[[i]])
}
```

###Results

Save 

```{r}
for (i in 1:length(ort.ant.sb.res)){
  ort.ant.sb.res[[i]]$Contrast <- paste(treats[i])
  ort.ant.sb.res[[i]]$OrthoGroup <- rownames(ort.ant.sb.res[[i]])
  rownames(ort.ant.sb.res[[i]]) <- NULL
  ort.ant.sb.res[[i]] <- as.data.frame(ort.ant.sb.res[[i]])
}

ort.ant.sb.res.all <- bind_rows(ort.ant.sb.res[1:3])

write.table(ort.ant.sb.res.all, paste(wd, "Anthophila_sB_DeSeqResults_All.tsv", sep = ""),
            col.names = T, row.names = F, quote = F, sep = "\t")

ort.ant.sb.res.sig <- subset(ort.ant.sb.res.all, padj < 0.1)

write.table(ort.ant.sb.res.sig, paste(wd,
                                      "Anthophila_sB_DeSeqResults_FDRlt0.1.tsv", sep =""), 
            col.names = T, row.names = F, quote = F, sep = "\t")
```

###ExpBoxPlots

What is signficant ? 

```{r}
ort.ant.sb.res.sig %>%
  arrange(OrthoGroup)
```


```{r}
sig.ort <- ort.ant.sb.res.sig$OrthoGroup
ann[ann$Orthogroup %in% sig.ort,] %>%
  arrange(Orthogroup)
```

Only labelling immune / putative immune genes.

```{r}
labs <- sig.ort
labs[labs=="OG0004091"] <- "Argo-2"
labs[labs=="OG0004253"] <- "Apoliphorins"
labs[labs=="OG0004610"] <- "Hymenoptaecin"
labs[labs=="OG0008153"] <- "Fibroin heavy chain"
labs[labs=="OG0008945"] <- "Cytochrome b5"
labs[labs=="OG0000635"] <- "Defensin"
labs[labs=="OG0000699"] <- "Cactus"
labs[labs=="OG0001255"] <- "Uncharacterised OG0001255"
labs[labs=="OG0003982"] <- "Leukocyte elastase inhibitor"
```


```{r}
for (i in 1:length(sig.ort)){
  visiOrt.messy(sig.ort[i], labs[i], ort.ant.sb.dds)
  visiOrt.clean(sig.ort[i], labs[i], ort.ant.sb.dds)
}
```


Make some space

```{r}
remove(ort.ant.sb.dds, ort.ant.sb.res, 
       ort.ant.sb.res.all, ort.ant.sb.res.sig, 
       ort.ant.sb.tx)
```



##OrthoLevel DE: All Bee (Anthophila) sans all outliers

###Preparing TX Object


```{r}
runs <- c("Amel", "Bter", "Caus")
filterCount(runs, "Anthophila_sA")
```

There's three ways to run this: with all species, with only Bter outliers removed, and with all putative outliers removed.

This time, I'll only remove Bter outliers.

```{r}
wd <- "output/Ortho_DE/Anthophila_sA/"
outliers <- c("BT_A_N_001", "BT_A_P_030", "BT_A_SL_003", "BT_A_SM_032",
              "CA_N_012", "CA_P_055", "CA_SL_056")
```


And now to make a list of txobjects (one per species) from these filtered count tables.

```{r}
ort.ant.sa.txs <- ortCount.outRem("Anthophila_sA", tx.list, outliers)

#make combined abundance matrix
abundance <- cbind(ort.ant.sa.txs[[1]]$abundance,
                   ort.ant.sa.txs[[2]]$abundance,
                   ort.ant.sa.txs[[3]]$abundance)
nrow(abundance)
```

```{r}
#make combined counts matrix
counts <- cbind(ort.ant.sa.txs[[1]]$counts,
                   ort.ant.sa.txs[[2]]$counts,
                   ort.ant.sa.txs[[3]]$counts)
nrow(counts)
```

```{r}
#make combined lengths matrix
length <- cbind(ort.ant.sa.txs[[1]]$length,
                   ort.ant.sa.txs[[2]]$length,
                   ort.ant.sa.txs[[3]]$length)
nrow(length)
```

```{r}
ort.ant.sa.tx <- list(abundance = abundance,
                  counts = counts, 
                  length = length)

ort.ant.sa.tx$countsFromAbundance <- "no"

for (i in 1:length(ort.ant.sa.tx)){
  print(head(ort.ant.sa.tx[[i]], n = 1))
}
```

```{r}
remove(ort.ant.sa.txs)
```


###DE Analysis

```{r}
ort.ant.sa.dds <- getDDS("Anthophila_sA", ort.ant.sa.tx)
```

```{r}
ort.ant.sa.res <- vector(mode = "list", length = 3)
treats <- resultsNames(ort.ant.sa.dds)[c(4,3,2)]

for(i in 1:length(ort.ant.sa.res)){
  ort.ant.sa.res[[i]] <- results(ort.ant.sa.dds, name = paste(treats[i]))
  print(treats[i])
  summary(ort.ant.sa.res[[i]])
}
```

###Results

Save 

```{r}
for (i in 1:length(ort.ant.sa.res)){
  ort.ant.sa.res[[i]]$Contrast <- paste(treats[i])
  ort.ant.sa.res[[i]]$OrthoGroup <- rownames(ort.ant.sa.res[[i]])
  rownames(ort.ant.sa.res[[i]]) <- NULL
  ort.ant.sa.res[[i]] <- as.data.frame(ort.ant.sa.res[[i]])
}

ort.ant.sa.res.all <- bind_rows(ort.ant.sa.res[1:3])

write.table(ort.ant.sa.res.all, paste(wd, "Anthophila_sA_DeSeqResults_All.tsv", sep = ""),
            col.names = T, row.names = F, quote = F, sep = "\t")

ort.ant.sa.res.sig <- subset(ort.ant.sa.res.all, padj < 0.1)

write.table(ort.ant.sa.res.sig, paste(wd,
                                      "Anthophila_sA_DeSeqResults_FDRlt0.1.tsv", sep =""), 
            col.names = T, row.names = F, quote = F, sep = "\t")
```

###ExpBoxPlots

What is signficant ? 

```{r}
ort.ant.sa.res.sig %>%
  arrange(OrthoGroup)
```


```{r}
sig.ort <- ort.ant.sa.res.sig$OrthoGroup
ann[ann$Orthogroup %in% sig.ort,] %>%
  arrange(Orthogroup)
```

Only labelling immune / putative immune genes.

```{r}
labs <- sig.ort
labs[labs=="OG0004091"] <- "Argo-2"
labs[labs=="OG0004253"] <- "Apoliphorins"
labs[labs=="OG0004610"] <- "Hymenoptaecin"
labs[labs=="OG0008153"] <- "Fibroin heavy chain"
labs[labs=="OG0008945"] <- "Cytochrome b5"
labs[labs=="OG0000635"] <- "Defensin"
labs[labs=="OG0000699"] <- "Cactus"
labs[labs=="OG0001255"] <- "Uncharacterised OG0001255"
labs[labs=="OG0003982"] <- "Leukocyte elastase inhibitor"
labs[labs=="OG0000399"] <- "Toll"
labs[labs=="OG0002872"] <- "CD109 antigen"
```


```{r}
for (i in 1:length(sig.ort)){
  visiOrt.messy(sig.ort[i], labs[i], ort.ant.sa.dds)
  visiOrt.clean(sig.ort[i], labs[i], ort.ant.sa.dds)
}
```


##OrthoLevel DE: Apis and Bombus (Apinae) sans Bter outliers

###Preparing TX Object

```{r}
runs <- c("Amel", "Bter")
filterCount(runs, "Apinae_sB")
```

There's three ways to run this: with all species, with only Bter outliers removed, and with all putative outliers removed.

This time, I'll only remove Bter outliers.

```{r}
wd <- "output/Ortho_DE/Apinae_sB/"
outliers <- c("BT_A_N_001", "BT_A_P_030", "BT_A_SL_003", "BT_A_SM_032")
```


And now to make a list of txobjects (one per species) from these filtered count tables.

```{r}
ort.api.sb.txs <- ortCount.outRem("Apinae_sB", tx.list, outliers)

#make combined abundance matrix
abundance <- cbind(ort.api.sb.txs[[1]]$abundance,
                   ort.api.sb.txs[[2]]$abundance)
nrow(abundance)
```

```{r}
#make combined counts matrix
counts <- cbind(ort.api.sb.txs[[1]]$counts,
                   ort.api.sb.txs[[2]]$counts)
nrow(counts)
```

```{r}
#make combined lengths matrix
length <- cbind(ort.api.sb.txs[[1]]$length,
                   ort.api.sb.txs[[2]]$length)
nrow(length)
```

```{r}
ort.api.sb.tx <- list(abundance = abundance,
                  counts = counts, 
                  length = length)

ort.api.sb.tx$countsFromAbundance <- "no"

for (i in 1:length(ort.api.sb.tx)){
  print(head(ort.api.sb.tx[[i]], n = 1))
}
```

```{r}
remove(ort.api.sb.txs)
```


###DE Analysis

```{r}
ort.api.sb.dds <- getDDS("Apinae_sB", ort.api.sb.tx)
```

```{r}
ort.api.sb.res <- vector(mode = "list", length = 3)
treats <- resultsNames(ort.api.sb.dds)[c(4,3,2)]

for(i in 1:length(ort.api.sb.res)){
  ort.api.sb.res[[i]] <- results(ort.api.sb.dds, name = paste(treats[i]))
  print(treats[i])
  summary(ort.api.sb.res[[i]])
}
```

###Results

Save 

```{r}
for (i in 1:length(ort.api.sb.res)){
  ort.api.sb.res[[i]]$Contrast <- paste(treats[i])
  ort.api.sb.res[[i]]$OrthoGroup <- rownames(ort.api.sb.res[[i]])
  rownames(ort.api.sb.res[[i]]) <- NULL
  ort.api.sb.res[[i]] <- as.data.frame(ort.api.sb.res[[i]])
}

ort.api.sb.res.all <- bind_rows(ort.api.sb.res[1:3])

write.table(ort.api.sb.res.all, paste(wd, "Apinae_sB_DeSeqResults_All.tsv", sep = ""),
            col.names = T, row.names = F, quote = F, sep = "\t")

ort.api.sb.res.sig <- subset(ort.api.sb.res.all, padj < 0.1)

write.table(ort.api.sb.res.sig, paste(wd,
                                      "Apinae_sB_DeSeqResults_FDRlt0.1.tsv", sep =""), 
            col.names = T, row.names = F, quote = F, sep = "\t")
```

###ExpBoxPlots

What is signficant ? 

```{r}
ort.api.sb.res.sig %>%
  arrange(OrthoGroup)
```


```{r}
sig.ort <- ort.api.sb.res.sig$OrthoGroup
ann[ann$Orthogroup %in% sig.ort,] %>%
  arrange(Orthogroup)
```

Only labelling immune / putative immune genes.

```{r}
labs <- sig.ort
labs[labs=="OG0004091"] <- "Argo-2"
labs[labs=="OG0004253"] <- "Apoliphorins"
labs[labs=="OG0004610"] <- "Hymenoptaecin"
labs[labs=="OG0008153"] <- "Fibroin heavy chain"
labs[labs=="OG0008945"] <- "Cytochrome b5"
labs[labs=="OG0000635"] <- "Defensin"
labs[labs=="OG0000699"] <- "Cactus"
labs[labs=="OG0001255"] <- "Uncharacterised OG0001255"
labs[labs=="OG0003982"] <- "Leukocyte elastase inhibitor"
labs[labs=="OG0000138"] <- "Modular serine protease"
labs[labs=="OG0000244"] <- "Uncharacterised OG0000244"
labs[labs=="OG0000399"] <- "Toll"
labs[labs=="OG0000446"] <- "Antitrypsin"
labs[labs=="OG0001412"] <- "PGRP"
labs[labs=="OG0001762"] <- "L-threonine ammonia-lyase"
labs[labs=="OG0001768"] <- "HSDL1"
labs[labs=="OG0002284"] <- "pug"
labs[labs=="OG0002872"] <- "CD109 antigen"
labs[labs=="OG0003154"] <- "Hox-B1a"
labs[labs=="OG0004253"] <- "Apolipophorins"
labs[labs=="OG0004983"] <- "Serpin 88Ea"
labs[labs=="OG0006763"] <- "Loricrin"
labs[labs=="OG0007796"] <- "Myo-inositol 2-dehydrogenase"
labs[labs=="OG0008950"] <- "Cytochrome b561 domain-containing protein 2"
labs[labs=="OG0008363"] <- "Inositol oxygenase"
labs[labs=="OG0009073"] <- "GCWSP"
labs[labs=="OG0009885"] <- "Uncharacterised OG0009885"
```


```{r}
for (i in 1:length(sig.ort)){
  visiOrt.messy(sig.ort[i], labs[i], ort.api.sb.dds)
  visiOrt.clean(sig.ort[i], labs[i], ort.api.sb.dds)
}
```


Make some space

```{r}
remove(ort.api.sb.dds, ort.api.sb.res, 
       ort.api.sb.res.all, ort.api.sb.res.sig, 
       ort.api.sb.tx)
```


##SessionInfo

```{r}
writeLines(capture.output(sessionInfo()),
           "SessionLogs/Proj_Pt5_SesInfo.txt")
sessionInfo()
```