---
title: "Comp_Hym_Project"
output: html_document
---

12th July 2022

Gene level differential expression analysis.

```{r}
set.seed(42)
dir.create("output")
dir.create("output/Ind_DE/")
```


##Libraries

```{r}
library("DESeq2")
library("tidyverse")
library("ggrepel")
library("tximport")
library("pheatmap")
library("reshape")
library("ggvenn")
```

##Resources

Following recommended workflow: http://master.bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html#preparing-quantification-input-to-deseq2

DeSeq2: https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html

Trim Galore:  https://github.com/FelixKrueger/TrimGalore

Kallisto :    https://pachterlab.github.io/kallisto/about

Gene level instructions from tximport: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4712774/

OrthoGroup information / Annotation information

```{r}
ortho <- read.table("input/Ortho_Files/OrthoFinder_Combined_Parsed_Jun22.tsv",
                    sep = "\t", header = T)

ann <- read.table("input/Gene_Info/Amel_ImmAnnotations_Orthogroup_Jun22.tsv",
                  header = T, sep = "\t")
ann.slim <- ann[,c(2:3)]
```


##Experiment Notes 

Previously, a standardised immune stimulation protocol was used in four different Hymenoptera species of varying sociality: Apis mellifera, Bombus terrestris, Ceratina calcarata and Polistes canadensis (three bees and a wasp, respectively.) The experiment was comprised of four treatments: 

1. Naive (N), where the animals were handled much as the others were but no treatment was administered.

2. Sterile stab / PBS (P), where the animals had their abdomens broken with an insect needle dipped in sterile PBS solution.

3. Gram -ve / Staph (SL), where the animals had their abdomens broken with an insect needle dipped in culture containing heat-killed Staphyloccus lentus.

4. Gram +ve / Serr (SM), where the animals had their abdomens broken with an insect needle dipped in culture containing heat-killed Serratia marascens.

All "injections" were administered on the dorsal side of the abdomen. Animals were then kept for 6 hours before being snap-frozen in liquid nitrogen. Abdomens were homogenized in TRIzol using a tissue lyser and chloroform was applied to allow for phase separation. The aqueous was then combined with equal volume ethanol (100%) and RNA was then extracted using Zymo Direct-zol MiniPrep kits.

Library prep and sequencing were undergone out of house using Novogene (details to be added)

##The Bioinformatics

All of the below mentioned files are found in the Scripts/Bash/ directory of this project folder. This work was done using multiple threads on a server.

Each of the data packets were uncompressed and checked against their provided MD5.txt files to check for possible corruption (MDcheck.sh). When all samples passed this step, the read files were trimmed using Trim Galore wrapper (https://github.com/FelixKrueger/TrimGalore). Genome indices and mapping was undergone using Kallisto (https://pachterlab.github.io/kallisto/about). Genomes were indexed manually and mapping was automated using a bash script (MapKal_v3.sh). 

The results for each sample were ordered by species and currently reside in input/Counts_Kal/

Below is the average % mapped of the K-mers per species

```{r}
lg <- read.table("input/logs_Kal/PercMapped.tsv", header = T, sep = "\t")
lg2 <- lg %>% 
        group_by(SampleSet) %>%
        summarise(PercMapped = mean(PercMapped)) 
lg2
```

##Functions

Converts transcript-level counts from input/ directory to a single counts table with gene level estimates. Also generates a coldata object in the output directory

```{r}
geneCount <- function(species){
  dir <- paste("input/Counts_Kal/", species, sep = "")
  files <- list.files(dir)
  samples <- sapply(strsplit(files, ".", fixed = T), "[", 1)
  treat <- c(rep("Naive", 3), rep("Wound", 3), rep("GramPos", 3),
             rep("GramNeg",3))
  coldata <- as.data.frame(cbind(samples, treat))
  names(coldata) <- c("Sample", "Treatment")
  write.table(coldata, 
              paste("output/Ind_DE/", species, "/", species, "_SampleData.tsv", sep = ""),
              col.names = T, row.names = F, sep = "\t", quote = F)
  names(files) <- samples
  txgene <- read.table(paste("input/Gene_Info/", species, "_TXGene.tsv", sep =""),
                       header = T, sep = "\t")
  cnts <- file.path(dir, files)
  names(cnts) <- samples
  txi.kal.tsv <- tximport(cnts, type = "kallisto", tx2gene = txgene, ignoreAfterBar = T)
  write.table(txi.kal.tsv$counts,
              paste("output/Ind_DE/", species, "/", 
                    species,  "_TxGeneCount.tsv", sep = ""),
              col.names = T, row.names = T, quote = F, sep = "\t")
  return(txi.kal.tsv)
}
```

Produce DDS object from tximport object (includes prefiltering and setting reference condition to Naive)

```{r}
getDDS <- function(species, txobject){
  cd <- read.table(paste("output/Ind_DE/", species, "/",
                         species, "_SampleData.tsv", sep = ""), header = T, sep = "\t")
  ddsTx <- DESeqDataSetFromTximport(txobject, colData = cd, design = ~Treatment)
  keep <- rowSums(counts(ddsTx)) >= 10
  ddsTx2 <- ddsTx[keep,]
  ddsTx2$Treatment <- relevel(ddsTx2$Treatment, ref = "Naive")
  ddsTx2 <- DESeq(ddsTx2)
  return(ddsTx2)
}
```

Visually assess the samples if (an) outlier(s) was/were removed

```{r}
outCheck <- function(outlier_s, species){
  cd <- read.table(paste("output/Ind_DE/", species, "/",
                         species, "_SampleData.tsv", sep = ""), header = T)
  cd <- cd[!cd$Sample %in% outlier_s,]
  dir <- paste("input/Counts_Kal/", species, sep = "")
  files <- list.files(dir)
  samples <- sapply(strsplit(files, ".", fixed = T), "[", 1)
  txgene <- read.table(paste("input/Gene_Info/", species, "_TXGene.tsv", sep =""),
                         header = T, sep = "\t")
  cnts <- file.path(dir, files)
  names(cnts) <- samples
  cnts[outlier_s]
  cnts <- cnts[!names(cnts) %in% outlier_s]
  tx <- tximport(cnts, type = "kallisto", tx2gene = txgene, ignoreAfterBar = T)
  
  ddsTx <- DESeqDataSetFromTximport(tx, colData = cd, design = ~Treatment)
  keep <- rowSums(counts(ddsTx)) >= 10
  ddsTx2 <- ddsTx[keep,]
  ddsTx2$Treatment <- relevel(ddsTx2$Treatment, ref = "Naive")
  ddsTx2 <- DESeq(ddsTx2)
  vsd <- vst(ddsTx2, blind = F)
  select <- order(rowMeans(counts(ddsTx2,normalized=TRUE)),
                decreasing=TRUE)[1:20]
  p1 <- pheatmap(assay(vsd)[select,])
  sampleDists <- dist(t(assay(vsd)))
  sampleDistMatrix <- as.matrix(sampleDists)
  rownames(sampleDistMatrix) <- paste(vsd$Treatment, vsd$type, sep="-")
  colnames(sampleDistMatrix) <- paste(vsd$Treatment, vsd$type, sep="-")
  p2 <- pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)
  pca <- plotPCA(vsd, intgroup="Treatment", returnData = TRUE)
  p3 <- ggplot(pca, aes(x = PC1, y = PC2, color = Treatment, label = name)) +
    geom_point() +
    geom_text()
  p1
  p2 
  p3
}
```

Return a dds object with specified outlier removed

```{r}
ddsOmit <- function(outlier_s, species){
 cd <- read.table(paste("output/Ind_DE/", species, "/",
                         species, "_SampleData.tsv", sep = ""), header = T)
  cd <- cd[!cd$Sample %in% outlier_s,]
  dir <- paste("input/Counts_Kal/", species, sep = "")
  files <- list.files(dir)
  samples <- sapply(strsplit(files, ".", fixed = T), "[", 1)
  txgene <- read.table(paste("input/Gene_Info/", species, "_TXGene.tsv", sep =""),
                         header = T, sep = "\t")
  cnts <- file.path(dir, files)
  names(cnts) <- samples
  cnts[outlier_s]
  cnts <- cnts[!names(cnts) %in% outlier_s]
  tx <- tximport(cnts, type = "kallisto", tx2gene = txgene, ignoreAfterBar = T)
  
  ddsTx <- DESeqDataSetFromTximport(tx, colData = cd, design = ~Treatment)
  keep <- rowSums(counts(ddsTx)) >= 10
  ddsTx2 <- ddsTx[keep,]
  ddsTx2$Treatment <- relevel(ddsTx2$Treatment, ref = "Naive")
  ddsTx2 <- DESeq(ddsTx2)
  return(ddsTx2)
}
```

Produce a dataframe with orthogroup and gene class information ready for data visualisation

```{r}
makeMetaDf <- function(res.all.object){
  x <- res.all.object[!is.na(res.all.object$padj),]
  x$Condition[x$Contrast==paste(treats[1])] <- "Wound"
  x$Condition[x$Contrast==paste(treats[2])] <- "Gram Positive"
  x$Condition[x$Contrast==paste(treats[3])] <- "Gram Negative"
  for(i in 1:nrow(x)){
  check <- x$Gene[i] %in% ortho$GeneID
    if (check == FALSE){
      x$OrthoGroup[i] <- "Unassigned"
    } else {
    x$OrthoGroup[i] <- ortho$Orthogroup[ortho$GeneID ==
                                                       x$Gene[i]]
    }
  }
  for(i in 1:nrow(x)){
  if (x$OrthoGroup[i] == "Unassigned") {
    x$Function[i] <- "Unknown"} else {
      check2 <- x$OrthoGroup[i] %in% ann.slim$Orthogroup
      if (check2 == FALSE){
        x$Function[i] <- "Unknown"
      } else {
    f <- unique(ann.slim$Function[ann.slim$Orthogroup == x$OrthoGroup[i]])
    if (length(f) > 1){
      if(TRUE %in% grepl("Effector|Recognition|Signalling", f)){
        x$Function[i] <- "Immune"
    } else {
      if(TRUE %in% grepl("Putative Immune", f)){
        x$Function[i] <- "Putative Immune"
        } 
      }
    } else {
    f2 <- gsub("Effector", "Immune", 
               gsub("Recognition", "Immune", 
                    gsub("Signalling", "Immune", 
                         gsub("Background", "None Immune", f))))
    x$Function[i] <- paste(f2)
        }
      }
    }
  }
  x$Sig <- "No"
  x$Sig[x$padj < 0.1] <- "Yes"
  x$Condition <- factor(x$Condition, levels = c("Wound", 
                                                "Gram Negative", 
                                                "Gram Positive"))
  x$Neglog10adjPvalue <- -log10(x$padj)
  x$Direction <- "Not Significant"
  x$Direction[x$Sig == "Yes" &
                    x$log2FoldChange > 0 ] <- "Up-Regulated"
  x$Direction[x$Sig == "Yes" &
                    x$log2FoldChange < 0 ] <- "Down-Regulated"
  return(x)
}
```

Produce a 1/0 binary matrix for use in comparing overlap in genes of interest

```{r}
makeMatrix <- function(goidf){
  mat <- as.data.frame(unique(goidf$goi))
  names(mat) <- "Gene"
  mat$SigWound <- ifelse(mat$Gene %in% sig.wound, 1, 0)
  mat$SigPos <- ifelse(mat$Gene %in% sig.pos, 1, 0)
  mat$SigNeg <- ifelse(mat$Gene %in% sig.neg, 1, 0)
  mat$TopWound <- ifelse(mat$Gene %in% top.wound, 1, 0)
  mat$TopPos <- ifelse(mat$Gene %in% top.pos, 1, 0)
  mat$TopNeg <- ifelse(mat$Gene %in% top.neg, 1, 0)
  mat$AllSig <- ifelse(mat$SigWound == 1 &
                      mat$SigPos == 1 &
                      mat$SigNeg == 1, 1, 0)
  mat$AllTop <- ifelse(mat$TopWound == 1 &
                      mat$TopPos == 1 &
                      mat$TopNeg == 1, 1, 0)
  mat$BacterialSig <- ifelse(mat$SigWound == 0 &
                      mat$SigPos == 1 &
                      mat$SigNeg == 1, 1, 0)
  mat$BacterialTop <- ifelse(mat$TopWound == 0 &
                      mat$TopPos == 1 &
                      mat$TopNeg == 1, 1, 0)
  mat$GoiWound <- ifelse(mat$TopWound == 1 | mat$SigWound == 1, 1, 0)
  mat$GoiPos <- ifelse(mat$TopPos == 1 | mat$SigPos == 1, 1, 0)
  mat$GoiNeg <- ifelse(mat$TopNeg == 1 | mat$SigNeg == 1, 1, 0)
  mat$AllGoi <- ifelse(mat$GoiWound == 1 &
                      mat$GoiPos == 1 &
                        mat$GoiNeg == 1, 1, 0)
  mat$BacterialGoi <- ifelse(mat$GoiPos == 1 &
                        mat$GoiNeg == 1  &
                          mat$GoiWound == 0, 1, 0)
  return(mat)
}
```

Produce and save a boxplot with expression levels across samples with gene input

```{r}
visiGene <- function(gene, genename, dds){
  tmp <- counts(dds, normalized = T)
  samples <- dds$Sample
  counts <- tmp[rownames(tmp) == paste(gene),]
  plot.df <- as.data.frame(cbind(samples, counts))
  names(plot.df) <- c("Samples", "Norm_Counts")
  plot.df$Treatment[grepl("_N_", plot.df$Samples)] <- "Naive"
  plot.df$Treatment[grepl("_P_", plot.df$Samples)] <- "Wound"
  plot.df$Treatment[grepl("_SM_", plot.df$Samples)] <- "Gram Negative"
  plot.df$Treatment[grepl("_SL_", plot.df$Samples)] <- "Gram Positive"
  plot.df$Treatment <- factor(plot.df$Treatment, levels = c("Naive",
                                                        "Wound", 
                                                        "Gram Positive", 
                                                        "Gram Negative"))
  plot.df$Norm_Counts <- as.numeric(plot.df$Norm_Counts)
  p <- ggplot(plot.df, aes(x = Treatment, y = log(Norm_Counts), fill = Treatment)) +
    geom_boxplot() +
    geom_point(position = position_jitter(w=0.1, h = 0),
               aes(colour = Treatment), size = 3) +
    scale_colour_manual(values = c("#58ffa7", "#ffa758", "#ff58b0", "#58b0ff")) +
    xlab("Treatment") +
    labs(title = (paste(genename, " expression across treatments", sep = ""))) +
    guides(alpha = FALSE, colour = guide_legend("Treatment"))
  ggsave(paste(wd, "Plots/ExpressionPlots/", gene, "_ExpBoxPlot.pdf", sep = ""))
  p
}
```

Plot all expression boxplots from a group of genes 

```{r}
visiFacet <- function(genes, genenames, dds, filename){
  p <- vector(mode = "list", length = length(genes))
  for (i in 1:length(p)){
    tmp <- counts(dds, normalized = T)
    samples <- dds$Sample
    counts <- tmp[rownames(tmp) == paste(genes[i]),]
    p[[i]] <- as.data.frame(cbind(samples, counts))
    names(p[[i]]) <- c("Samples", "Norm_Counts")
    p[[i]]$Treatment[grepl("_N_", p[[i]]$Samples)] <- "Naive"
    p[[i]]$Treatment[grepl("_P_", p[[i]]$Samples)] <- "Wound"
    p[[i]]$Treatment[grepl("_SM_", p[[i]]$Samples)] <- "Gram Negative"
    p[[i]]$Treatment[grepl("_SL_", p[[i]]$Samples)] <- "Gram Positive"
    p[[i]]$Treatment <- factor(p[[i]]$Treatment, levels = c("Naive",
                                                          "Wound", 
                                                          "Gram Positive", 
                                                          "Gram Negative"))
    p[[i]]$Norm_Counts <- as.numeric(p[[i]]$Norm_Counts)
    p[[i]]$Gene <- paste(genes[i])
    p[[i]]$Label <- paste(genenames[i])
  }
  p2 <- bind_rows(p)
  p3 <- ggplot(p2, aes(x = Treatment, y = log(Norm_Counts))) +
    geom_boxplot(aes(colour = Treatment, alpha = 0.5)) +
    geom_point(position = position_jitter(w=0.1, h = 0),
               aes(colour = Treatment), size = 3) +
    scale_colour_manual(values = c("#58ffa7", "#ffa758", "#ff58b0", "#58b0ff")) +
    xlab("Treatment") +
    guides(alpha = FALSE, colour = guide_legend("Treatment")) +
    facet_wrap(~Label, nrow = 5, scale = "free_y")
  ggsave(paste(wd, "Plots/ExpressionPlots/", filename, "_Facet_ExpBoxPlot.pdf", sep = ""))
  p3
}
```



##Amel: Gene Level DE 

```{r}
dir.create("output/Ind_DE/Amel/")
wd <- "output/Ind_DE/Amel/"
```


###DE Analysis

First, convert transcript-level counts table from Kallisto to gene level counts and produce a coldata object

```{r}
amel.tx <- geneCount("Amel")
head(amel.tx$counts)
```

Build a DESeq DataSet from the above, prefilter (keep only rows that have at least 10 reads total), and return dds object.

```{r}
amel.dds <- getDDS("Amel", amel.tx)
amel.res <- vector(mode = "list", length = 3)
treats <- resultsNames(amel.dds)[c(4,3,2)]

for(i in 1:length(amel.res)){
  amel.res[[i]] <- results(amel.dds, name = paste(treats[i]))
  summary(amel.res[[i]])
}
```

###Check for outliers

First, check counts

```{r}
boxplot(log10(assays(amel.dds)[["cooks"]]), range=0, las=2)
```

```{r}
#normalise
vsd <- vst(amel.dds, blind = F)
#make heatmap (top 20 genes)
select <- order(rowMeans(counts(amel.dds,normalized=TRUE)),
                decreasing=TRUE)[1:20]
pheatmap(assay(vsd)[select,])
```

```{r}
pdf("output/Ind_DE/Amel/Amel_Pheatmap.pdf")
  pheatmap(assay(vsd)[select,])
dev.off()
```


AM_N_056 is clustering with bacterial and wound challenges... which may be a problem.

Check with sample to sample distances

```{r}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$Treatment, vsd$type, sep="-")
colnames(sampleDistMatrix) <- NULL

pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)
```
That outlying Naive is a log way from the others ... 

```{r}
pdf("output/Ind_DE/Amel/Amel_DistanceMatrix.pdf")
  pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)
dev.off()
```

Finally, PCA

```{r}
amel.pca <- plotPCA(vsd, intgroup="Treatment", returnData = TRUE)
ggplot(amel.pca, aes(x = PC1, y = PC2, color = Treatment, label = name)) +
  geom_point() +
  geom_text()
```
```{r}
ggsave("output/Ind_DE/Amel/Amel_PCA.pdf")
```

What happens if I remove the potential outlier ? 

```{r}
torem <- "AM_N_056"
outCheck(torem, "Amel")
```

I'm not convinced that I need to remove that sample, but I'll look at the results regardless.

```{r}
amel.dds2 <- ddsOmit(torem, "Amel")

amel.res2 <- vector(mode = "list", length = 3)
treats2 <- resultsNames(amel.dds2)[c(4,3,2)]

for(i in 1:length(amel.res2)){
  amel.res2[[i]] <- results(amel.dds2, name = paste(treats2[i]))
  print(treats2[i])
  summary(amel.res2[[i]])
}
```

Save plots

```{r}
vsd2 <- vst(amel.dds2, blind = F)
#make heatmap (top 20 genes)
select <- order(rowMeans(counts(amel.dds2,normalized=TRUE)),
                decreasing=TRUE)[1:20]
pheatmap(assay(vsd2)[select,])
```

```{r}
pdf("output/Ind_DE/Amel/Amel_OutlierRemoved_Pheatmap.pdf")
  pheatmap(assay(vsd2)[select,])
dev.off()
```

```{r}
sampleDists2 <- dist(t(assay(vsd2)))
sampleDistMatrix2 <- as.matrix(sampleDists2)
rownames(sampleDistMatrix2) <- paste(vsd2$Treatment, vsd2$type, sep="-")
colnames(sampleDistMatrix2) <- NULL

pheatmap(sampleDistMatrix2,
         clustering_distance_rows=sampleDists2,
         clustering_distance_cols=sampleDists2)
```

```{r}
pdf("output/Ind_DE/Amel/Amel_OutlierRemoved_DistanceMatrix.pdf")
  pheatmap(sampleDistMatrix2,
         clustering_distance_rows=sampleDists2,
         clustering_distance_cols=sampleDists2)
dev.off()
```


```{r}
amel.pca2 <- plotPCA(vsd2, intgroup="Treatment", returnData = TRUE)
ggplot(amel.pca2, aes(x = PC1, y = PC2, color = Treatment, label = name)) +
  geom_point() +
  geom_text()
ggsave("output/Ind_DE/Amel/Amel_OutlierRemoved_PCA.pdf")
```


Do we see a difference in number of sig genes in terms of whether they're immune or not ? 

```{r}
sig <- vector(mode = "list", length = length(amel.res))

for (i in 1:length(sig)){
  tmp <- amel.res[[i]][!is.na(amel.res),]
  tmp$Gene <- rownames(tmp)
  sig[[i]] <- tmp$Gene[tmp$padj < 0.1]
  sig[[i]] <- sig[[i]][!is.na(sig[[i]])]
}

run1 <- unique(unlist(sig))
ort1 <- vector(length = length(run1))
ann1 <- vector(length = length(run1))

for(i in 1:length(run1)){
  check <- run1[i] %in% ortho$GeneID
  if (check == FALSE){
    ort1[i] <- "Unassigned"
  } else {
  ort1[i] <- ortho$Orthogroup[ortho$GeneID == run1[i]]
  }
  ann1[i] <- ann$Function[ann$GeneID == run1[i]]
}

one <- as.data.frame(cbind(run1, ort1, ann1))
names(one) <- c("Gene", "OrthoGroup", "Function")
one$Run <- paste("One")
```

And again...

```{r}
sig2 <- vector(mode = "list", length = length(amel.res2))

for (i in 1:length(sig2)){
  tmp <- amel.res2[[i]][!is.na(amel.res2),]
  tmp$Gene <- rownames(tmp)
  sig2[[i]] <- tmp$Gene[tmp$padj < 0.1]
  sig2[[i]] <- sig2[[i]][!is.na(sig2[[i]])]
}

run2 <- unique(unlist(sig2))
ort2 <- vector(length = length(run2))
ann2 <- vector(length = length(run2))

for(i in 1:length(run2)){
  check <- run2[i] %in% ortho$GeneID
  if (check == FALSE){
    ort2[i] <- "Unassigned"
  } else {
  ort2[i] <- ortho$Orthogroup[ortho$GeneID == run2[i]]
  }
  ann2[i] <- ann$Function[ann$GeneID == run2[i]]
}

two <- as.data.frame(cbind(run2, ort2, ann2))
names(two) <- c("Gene", "OrthoGroup", "Function")
two$Run <- paste("Two")
```


```{r}
one <- as.data.frame(table(one$Function, one$Run))
one$Prop <- one$Freq/sum(one$Freq)
two <- as.data.frame(table(two$Function, two$Run))
two$Prop <- two$Freq/sum(two$Freq)

check.df <- rbind(one,two)

ggplot(check.df, aes(fill=Var1, y=Prop, x=Var2)) + 
    geom_bar(position="dodge", stat="identity")
```
No real difference. Gonna leave it as it is and not remove any outliers.

###Results

Save 

```{r}
for (i in 1:length(amel.res)){
  amel.res[[i]]$Contrast <- paste(treats[i])
  amel.res[[i]]$Gene <- rownames(amel.res[[i]])
  rownames(amel.res[[i]]) <- NULL
  amel.res[[i]] <- as.data.frame(amel.res[[i]])
}
 
amel.res.all <- bind_rows(amel.res[1:3])
 
id <- read.table("input/Gene_Info/Amel_IDwDesc.tsv", 
                 sep = "\t", quote = "", header = T)

for (i in 1:nrow(amel.res.all)){
  d <- unique(id$Description[id$GeneID == amel.res.all$Gene[i]])
  if(length(d) > 1){
    amel.res.all$Desc[i] <- d[1]
  } else {
    amel.res.all$Desc[i] <- d  
  }
}

write.table(amel.res.all, paste(wd, "Amel_DeSeqResults_All.tsv", sep = ""),
            col.names = T, row.names = F, quote = F, sep = "\t")

amel.res.sig <- subset(amel.res.all, padj < 0.1)

write.table(amel.res.sig, paste(wd, "Amel_DeSeqResults_FDRlt0.1.tsv", sep =""), 
            col.names = T, row.names = F, quote = F, sep = "\t")
```


Genes of interest?

I'm going to take the top 10 most significant and top 10 biggest logFC per treatment. 
Genes that overlap between two (and between the three treatments) will be of particular interest

```{r}
nona <- amel.res.all[!is.na(amel.res.all$padj),]
nona$LogDif <- sub('-', '', nona$log2FoldChange)
tmp <- nona[nona$Contrast == treats[1],]
sig.wound <- head(tmp$Gene[order(tmp$padj)], n = 10)
top.wound <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)
tmp <- nona[nona$Contrast == treats[2],]
sig.pos <- head(tmp$Gene[order(tmp$padj)], n = 10)
top.pos <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)
tmp <- nona[nona$Contrast == treats[3],]
sig.neg <-  head(tmp$Gene[order(tmp$padj)], n = 10)
top.neg <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)

goi <- c(sig.wound, top.wound, sig.pos, top.pos, sig.neg, top.neg)
stat <- c(rep("Top 10 Sig Wound", 10), rep("Top 10 log2FC Wound", 10),
          rep("Top 10 Sig Gram Positive", 10), rep("Top 10 log2FC Gram Positive", 10),
          rep("Top 10 Sig Gram Negative", 10), rep("Top 10 log2FC Gram Negative", 10))

goi.df <- as.data.frame(cbind(goi, stat))


goi.wound <- unique(c(sig.wound, top.wound))
goi.pos <- unique(c(sig.pos, top.pos))
goi.neg <- unique(c(sig.neg, top.neg))
```

###Visualise: Venn Diagrams

```{r}
top.sig <- list("Wound" = sig.wound,
                 "Gram Positive" = sig.pos,
                 "Gram Negative" = sig.neg)

p <- ggvenn(top.sig,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Top 10 Genes per Treatment by adjPvalue")
```
```{r}
dir.create(paste(wd, "Plots/", sep = ""))
ggsave(paste(wd, "Plots/Amel_TopAdjPvalue_Venn.pdf", sep = ""))
```

```{r}
top.lfc <- list("Wound" = top.wound,
                 "Gram Positive" = top.pos,
                 "Gram Negative" = top.neg)

p <- ggvenn(top.lfc,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Top 10 Genes per Treatment by log2FC")
```

```{r}
ggsave(paste(wd, "Plots/Amel_Toplog2FC_Venn.pdf", sep = ""))
```

```{r}
goi <- list("Wound" = goi.wound,
                 "Gram Positive" = goi.pos,
                 "Gram Negative" = goi.neg)

p <- ggvenn(goi,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Genes of Interest Across Treatments")
```

```{r}
ggsave(paste(wd, "Plots/Amel_GOI_Venn.pdf", sep = ""))
```

```{r}
goi.mat <- makeMatrix(goi.df)
goi.mat
```
```{r}
goi.mat$Gene[goi.mat$AllGoi == 1]
```

Cytochrome b5, alpha-mannosidase 2, fibroin heavy chain, defensin - 2, uncharacterized LOC408494

```{r}
goi.mat$Gene[goi.mat$BacterialGoi == 1]
```

NF-kappa-B inhibitor cactus(cactus2), dual 3',5'-cyclic-AMP and -GMP phosphodiesterase 11

```{r}
goi.mat$Gene[goi.mat$AllSig == 1]
```

Cytochrome b5, alpha-mannosidase 2, fibroin heavy chain

```{r}
goi.mat$Gene[goi.mat$AllTop == 1]
```

Defensin-2 and uncharacterized LOC408494

###Visualise: Expression BoxPlots

```{r}
dir.create(paste(wd, "Plots/ExpressionPlots/", sep = ""))
```

####Significant Wound

```{r}
goi.mat$Gene[goi.mat$SigWound == 1]
genes <- goi.mat$Gene[goi.mat$SigWound == 1]
```

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = amel.dds, 
          filename = "Woundlog2FC")
```

All Wound: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = amel.dds, 
          filename = "WoundSig")
```


####TopFC Wound

!!!! gonna stop with the gene by gene annotation, its too slow and changes as the analysis evolves. I

```{r}
goi.mat$Gene[goi.mat$TopWound == 1]
genes <- goi.mat$Gene[goi.mat$TopWound == 1]
```


All Wound: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = amel.dds, 
          filename = "Woundlog2FC")
```


####Significant GramPos

```{r}
goi.mat$Gene[goi.mat$SigPos == 1]
genes <- goi.mat$Gene[goi.mat$SigPos == 1]
```

All Gram Pos Sig: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = amel.dds, 
          filename = "PosSig")
```


####TopFC GramPos


```{r}
goi.mat$Gene[goi.mat$TopPos == 1]
genes <- goi.mat$Gene[goi.mat$TopPos == 1]
```



All log Gram Pos: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = amel.dds, 
          filename = "Poslog2FC")
```

####Significant GramNeg


```{r}
goi.mat$Gene[goi.mat$SigNeg == 1]
genes <- goi.mat$Gene[goi.mat$SigNeg == 1]
```


All Gram Neg Sig: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = amel.dds, 
          filename = "NegSig")
```

####TopFC GramPos


```{r}
goi.mat$Gene[goi.mat$TopNeg == 1]
genes <- goi.mat$Gene[goi.mat$TopNeg == 1]
```


###Visualise: PCA

Decided to not remove potential outliers.

Resultant PCA:

```{r}
ggplot(amel.pca, aes(x = PC1, y = PC2, color = Treatment)) +
  geom_point() 
```

```{r}
ggsave(paste(wd, "Plots/Amel_PCA_Simple.pdf", sep = ""))
```

```{r}
ggplot(amel.pca, aes(x = PC1, y = PC2, color = Treatment)) +
  geom_point() +
  geom_label_repel(
    data = amel.pca,
    aes(label = name),
    size = 3
  )
```
```{r}
ggsave(paste(wd, "Plots/Amel_PCA_Labelled.pdf", sep = ""))
```

```{r}
amel.pca$Treat2 <- c(rep("Naive", 3), rep("Treatment", 9))

ggplot(amel.pca, aes(x = PC1, y = PC2, color = Treat2)) +
  geom_point() +
  stat_ellipse()
```

```{r}
ggsave(paste(wd, "Plots/Amel_PCA_TrtVUnTrt.pdf", sep = ""))
```

###Visualise: Smears

Amel/Bter consideration : there is a shared gene name that causes issues. To fix:

```{r}
amel.res.all$Gene[amel.res.all$Gene == "Kr-h1"] <- "Kr-h1_Amel"

ortho$GeneID[ortho$GeneID == "Kr-h1" & ortho$Species == "Amel"] <- "Kr-h1_Amel"
```

```{r}
plot.df <- makeMetaDf(amel.res.all)
plot.df
```

The function assigns the genes without an orthogroup "unknown" functions. This is fine for the non Amel species but for Amel I can assign function based on gene ID

```{r}
for (i in 1:nrow(plot.df)){
  if (plot.df$Function[i] == "Unknown"){
    plot.df$Function[i] <- ann$Function[ann$GeneID == plot.df$Gene[i]]
  }
}

plot.df$Function <- gsub("Effector", "Immune", 
                      gsub("Recognition", "Immune", 
                      gsub("Signalling", "Immune", 
                      gsub("Background", "None Immune", plot.df$Function))))

tail(plot.df)
```


```{r}
plot.df$Condition <- factor(plot.df$Condition, levels = c("Wound",
                                                          "Gram Positive",
                                                          "Gram Negative"))

ggplot(plot.df, aes(x = log(baseMean), y = log2FoldChange)) +
  geom_point(aes(colour = Sig)) +
  facet_grid(~Condition) +
  scale_color_manual(values=c("black", "Red"), guide = NULL) +
  theme(legend.position = "bottom")
```
Note: remember only those that passed baseMean threshold per comparison are plotted (120 for Gram Pos, considerably lower for the others)

```{r}
ggsave(paste(wd, "Plots/Amel_SmearPlot.pdf", sep = ""))
```


###Visualise: Rainclouds

```{r}
sig.genes <- unique(unlist(sig))
plot.sig <- plot.df[plot.df$Gene %in% sig.genes,]

to.add.naive <- plot.sig[plot.sig$Gene %in% sig.genes &
                               plot.sig$Condition == "Wound",]
to.add.naive$Sig <- "No"
to.add.naive$Condition <- "Naive"
to.add.naive$log2FoldChange <- 0

plot.sig <- rbind(plot.sig, to.add.naive)

plot.sig$Condition <- factor(plot.sig$Condition, levels = c("Naive",
                                                            "Wound", 
                                                            "Gram Positive", 
                                                            "Gram Negative"))

plot.sig$Significance[plot.sig$padj > 0.1] <- "Not Significant"
plot.sig$Significance[plot.sig$padj < 0.1 & plot.sig$padj > 0.05] <- 
  "Significant (FDR < 0.1)"
plot.sig$Significance[plot.sig$padj < 0.05 & plot.sig$padj > 0.01] <-
  "Significant (FDR < 0.05)"
plot.sig$Significance[plot.sig$padj < 0.01 & plot.sig$padj > 0.001] <-
  "Significant (FDR < 0.01)"
plot.sig$Significance[plot.sig$padj < 0.001] <-
  "Significant (FDR < 0.001)"

plot.sig$Significance <- factor(plot.sig$Significance, levels =
                                 c("Not Significant",
                                   "Significant (FDR < 0.1)",
                                   "Significant (FDR < 0.05)",
                                   "Significant (FDR < 0.01)",
                                   "Significant (FDR < 0.001)"))


plot.sig$Function <- factor(plot.sig$Function, levels = c("Immune",
                                                          "Putative Immune", 
                                                          "None Immune"))


ggplot(plot.sig[!plot.sig$Condition == "Naive",], 
       (aes(x = Condition, y = log2FoldChange))) +
  geom_jitter(aes(colour = Function, shape = Significance)) +
  scale_colour_manual(values = c("#2c8cd9", "#7a2cd9", "darkgrey")) +
  scale_shape_manual(values = c(1, 5, 18, 19, 17)) 
```

```{r}
ggsave(paste(wd, "Plots/Amel_logFCJitter.pdf", sep = ""))
```

```{r}
ggplot(plot.sig, 
       (aes(x = Condition, y = log2FoldChange))) +
  geom_point(aes(colour = Function, shape = Significance)) +
  geom_line( linetype = "dashed", alpha = 0.2, aes(group = Gene, colour = Function)) +
  scale_colour_manual(values = c("#2c8cd9", "#7a2cd9", "darkgrey")) +
  scale_shape_manual(values = c(1, 5, 18, 19, 17)) 
```
```{r}
ggsave(paste(wd, "Plots/Amel_logFCScatter_GeneLine.pdf", sep = ""))
```


###Visualise: Volcano Plots

```{r}
plot.df$Function <- factor(plot.df$Function, levels = c("Immune",
                                                        "Putative Immune",
                                                        "None Immune"))

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.1", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```
```{r}
ggsave(paste(wd, "Plots/Amel_Simple_FDR0.1_Volcano.pdf", sep = ""))
```


```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1)) +
  guides(colour = guide_legend("FDR < 0.1", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```
```{r}
ggsave(paste(wd, "Plots/Amel_Class_FDR0.1_Volcano.pdf", sep = ""))
```

```{r}
plot.df$Direction2 <- "Not Significant"
plot.df$Direction2[plot.df$padj < 0.05 &
                    plot.df$log2FoldChange > 0 ] <- "Up-Regulated"
plot.df$Direction2[plot.df$padj < 0.05 &
                    plot.df$log2FoldChange < 0 ] <- "Down-Regulated"

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction2), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.05", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```
```{r}
ggsave(paste(wd, "Plots/Amel_Simple_FDR0.05_Volcano.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction2, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1)) +
  guides(colour = guide_legend("FDR < 0.05", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Amel_Class_FDR0.05_Volcano.pdf", sep = ""))
```


```{r}
plot.df$Direction3 <- "Not Significant"
plot.df$Direction3[plot.df$padj < 0.001 &
                    plot.df$log2FoldChange > 0 ] <- "Up-Regulated"
plot.df$Direction3[plot.df$padj < 0.001 &
                    plot.df$log2FoldChange < 0 ] <- "Down-Regulated"

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Amel_Simple_FDR0.001_Volcano.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1)) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Amel_Class_FDR0.001_Volcano.pdf", sep = ""))
```

Label top 3 sig genes per treatment

```{r}
sig.wound[1:3]
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "LOC724654"] <- "Cytochrome b5"
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "LOC102654628"] <- "Uncharacterised LOC102654628"
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "LOC113218757"] <- "LOC113218757"

sig.pos[1:3]
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "LOC726750"] <- "Fibroin heavy chain"
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "LOC411012"] <- "Cactus2"
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "LOC724654"] <- "Cytochrome b5"

sig.neg[1:3]
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "LOC726750"] <- "Fibroin heavy chain"
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "LOC411012"] <- "Cytochrome b5"
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "LOC411807"] <- "Uncharacterised LOC411807"
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") +
  geom_label_repel(
    data = plot.df[!is.na(plot.df$Label),],
    size = 2,
    aes(label = Label),
    point.padding = unit(0.3, "lines"),
    box.padding = 1.25
  ) 

```


```{r}
ggsave(paste(wd, "Plots/Amel_Simple_FDR0.001_Volcano_top3Label.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1)) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") +
  geom_label_repel(
    data = plot.df[!is.na(plot.df$Label),],
    size = 2,
    aes(label = Label),
    point.padding = unit(0.3, "lines"),
    box.padding = 1.2
  ) 

```
```{r}
ggsave(paste(wd, "Plots/Amel_Class_FDR0.001_Volcano_top3Label.pdf", sep = ""))
```


##Bter: Gene Level DE 

```{r}
dir.create("output/Ind_DE/Bter/")
wd <- "output/Ind_DE/Bter/"
```

NB: there are likely to be immune genes in Bter that do not group into orthogroups with Amel (ie rapidly evolving AMPs) and so they will have to be annotated separately.

LOC100649867: apidaecins type 73

```{r}
ortho %>%
  arrange(desc(Orthogroup))
```
```{r}
ortho[40557,1] <- "OG0011065"
ortho[40557,2] <- "Bter"
ortho[40557,3] <- "LOC100649867"

tail(ortho)
```
```{r}
ann.slim[12123, 1] <- "Effector"
ann.slim[12123, 2] <- "OG0011065"

tail(ann.slim)
```


###DE Analysis

First, convert transcript-level counts table from Kallisto to gene level counts and produce a coldata object

```{r}
bter.tx <- geneCount("Bter")
head(bter.tx$counts)
```

Build a DESeq DataSet from the above, prefilter (keep only rows that have at least 10 reads total), and return dds object.

```{r}
bter.dds <- getDDS("Bter", bter.tx)
bter.res <- vector(mode = "list", length = 3)
treats <- resultsNames(bter.dds)[c(4,3,2)]

for(i in 1:length(bter.res)){
  bter.res[[i]] <- results(bter.dds, name = paste(treats[i]))
  summary(bter.res[[i]])
}
```

###Check for outliers

First, check counts

```{r}
boxplot(log10(assays(bter.dds)[["cooks"]]), range=0, las=2)
```

```{r}
#normalise
vsd <- vst(bter.dds, blind = F)
#make heatmap (top 20 genes)
select <- order(rowMeans(counts(bter.dds,normalized=TRUE)),
                decreasing=TRUE)[1:20]
pheatmap(assay(vsd)[select,])
```
From this I would suggest P30 and SM32 are outliers - and possibly N001

```{r}
pdf("output/Ind_DE/Bter/Bter_Pheatmap.pdf")
  pheatmap(assay(vsd)[select,])
dev.off()
```

Check with sample to sample distances

```{r}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$Treatment, vsd$type, sep="-")
colnames(sampleDistMatrix) <- NULL

pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)
```

```{r}
pdf("output/Ind_DE/Bter/Bter_DistanceMatrix.pdf")
  pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)
dev.off()
```

Yeah that's pretty messy

Finally, PCA

```{r}
bter.pca <- plotPCA(vsd, intgroup="Treatment", returnData = TRUE)
ggplot(bter.pca, aes(x = PC1, y = PC2, color = Treatment, label = name)) +
  geom_point() +
  geom_text()
```
```{r}
ggsave("output/Ind_DE/Bter/Bter_PCA.pdf")
```

What happens if I remove the potential outliers ? 

```{r}
torem <- c("BT_A_SL_003", "BT_A_P_030", "BT_A_SM_032", "BT_A_N_001")
outCheck(torem, "Bter")
```

2 plots are still a mess, but the sample-distance plots are very much improved.
Now SL 003 is a problem ? Go back and try again....

Much better.... though I've lost 4 samples!


```{r}
bter.dds2 <- ddsOmit(torem, "Bter")

bter.res2 <- vector(mode = "list", length = 3)
treats2 <- resultsNames(bter.dds2)[c(4,3,2)]

for(i in 1:length(bter.res2)){
  bter.res2[[i]] <- results(bter.dds2, name = paste(treats2[i]))
  print(treats2[i])
  summary(bter.res2[[i]])
}
```

Save plots

```{r}
vsd2 <- vst(bter.dds2, blind = F)
#make heatmap (top 20 genes)
select <- order(rowMeans(counts(bter.dds2,normalized=TRUE)),
                decreasing=TRUE)[1:20]
pheatmap(assay(vsd2)[select,])
```

```{r}
pdf("output/Ind_DE/Bter/Bter_OutlierRemoved_Pheatmap.pdf")
  pheatmap(assay(vsd2)[select,])
dev.off()
```

```{r}
sampleDists2 <- dist(t(assay(vsd2)))
sampleDistMatrix2 <- as.matrix(sampleDists2)
rownames(sampleDistMatrix2) <- paste(vsd2$Treatment, vsd2$type, sep="-")
colnames(sampleDistMatrix2) <- NULL

pheatmap(sampleDistMatrix2,
         clustering_distance_rows=sampleDists2,
         clustering_distance_cols=sampleDists2)
```

```{r}
pdf("output/Ind_DE/Bter/Bter_OutlierRemoved_DistanceMatrix.pdf")
  pheatmap(sampleDistMatrix2,
         clustering_distance_rows=sampleDists2,
         clustering_distance_cols=sampleDists2)
dev.off()
```


```{r}
bter.pca2 <- plotPCA(vsd2, intgroup="Treatment", returnData = TRUE)
ggplot(bter.pca2, aes(x = PC1, y = PC2, color = Treatment, label = name)) +
  geom_point() +
  geom_text()
ggsave("output/Ind_DE/Bter/Bter_OutlierRemoved_PCA.pdf")
```

Do we see a difference in number of sig genes in terms of whether they're immune or not ? 

```{r}
sig <- vector(mode = "list", length = length(bter.res))

for (i in 1:length(sig)){
  tmp <- bter.res[[i]][!is.na(bter.res),]
  tmp$Gene <- rownames(tmp)
  sig[[i]] <- tmp$Gene[tmp$padj < 0.1]
  sig[[i]] <- sig[[i]][!is.na(sig[[i]])]
}

run1 <- unique(unlist(sig))
ort1 <- vector(length = length(run1))
ann1 <- vector(length = length(run1))

for(i in 1:length(run1)){
  check <- run1[i] %in% ortho$GeneID
  if (check == FALSE){
  ort1[i] <- "Unassigned"
  ann1[i] <- "Unknown"
} else {
  ort1[i] <- ortho$Orthogroup[ortho$GeneID == run1[i]]
  check2 <- ort1[i] %in% ann.slim$Orthogroup
  if (check2 == FALSE){
    ann1[i] <- "Unknown"
  } else {
  f <- unique(ann.slim$Function[ann.slim$Orthogroup == ort1[i]])
  if (length(f) > 1){
    if(TRUE %in% grepl("Effector|Recognition|Signalling", f)){
      ann1[i] <- "Immune"
    } else {
        if(TRUE %in% grepl("Putative Immune", f)){
          ann1[i] <- "Putative Immune"
        }
      }
  } else {
      f2 <- gsub("Effector", "Immune", 
               gsub("Recognition", "Immune", 
                    gsub("Signalling", "Immune", 
                         gsub("Background", "None Immune", f))))
      ann1[i] <- paste(f2)
      }
  }
  }
}

one <- as.data.frame(cbind(run1, ort1, ann1))
names(one) <- c("Gene", "OrthoGroup", "Function")
one$Run <- paste("One")
```

And again...

```{r}
sig2 <- vector(mode = "list", length = length(bter.res2))

for (i in 1:length(sig2)){
  tmp <- bter.res2[[i]][!is.na(bter.res2),]
  tmp$Gene <- rownames(tmp)
  sig2[[i]] <- tmp$Gene[tmp$padj < 0.1]
  sig2[[i]] <- sig2[[i]][!is.na(sig2[[i]])]
}

run2 <- unique(unlist(sig2))
ort2 <- vector(length = length(run2))
ann2 <- vector(length = length(run2))

for(i in 1:length(run2)){
  check <- run2[i] %in% ortho$GeneID
  if (check == FALSE){
  ort2[i] <- "Unassigned"
  ann2[i] <- "Unknown"
} else {
  ort2[i] <- ortho$Orthogroup[ortho$GeneID == run2[i]]
  check2 <- ort2[i] %in% ann.slim$Orthogroup
  if (check2 == FALSE){
    ann2[i] <- "Unknown"
  } else {
  f <- unique(ann.slim$Function[ann.slim$Orthogroup == ort2[i]])
  if (length(f) > 1){
    if(TRUE %in% grepl("Effector|Recognition|Signalling", f)){
      ann2[i] <- "Immune"
    } else {
        if(TRUE %in% grepl("Putative Immune", f)){
          ann2[i] <- "Putative Immune"
        }
      }
  } else {
      f2 <- gsub("Effector", "Immune", 
               gsub("Recognition", "Immune", 
                    gsub("Signalling", "Immune", 
                         gsub("Background", "None Immune", f))))
      ann2[i] <- paste(f2)
      }
  }
  }
}


two <- as.data.frame(cbind(run2, ort2, ann2))
names(two) <- c("Gene", "OrthoGroup", "Function")
two$Run <- paste("Two")
```


```{r}
one <- as.data.frame(table(one$Function, one$Run))
one$Prop <- one$Freq/sum(one$Freq)
two <- as.data.frame(table(two$Function, two$Run))
two$Prop <- two$Freq/sum(two$Freq)

check.df <- rbind(one,two)

ggplot(check.df, aes(fill=Var1, y=Prop, x=Var2)) + 
    geom_bar(position="dodge", stat="identity")
```

```{r}
ggplot(check.df, aes(fill=Var1, y=Freq, x=Var2)) + 
    geom_bar(position="dodge", stat="identity")
```

There is quite a significant difference .... gonna make the call to go forward without the outliers but .... this may change with Seth's input.

```{r}
bter.dds <- bter.dds2
bter.res <- bter.res2
```


###Results

Save 

(Once this is done once, read from the produced file, annotating takes some time.)

```{r}
for (i in 1:length(bter.res)){
  bter.res[[i]]$Contrast <- paste(treats[i])
  bter.res[[i]]$Gene <- rownames(bter.res[[i]])
  rownames(bter.res[[i]]) <- NULL
  bter.res[[i]] <- as.data.frame(bter.res[[i]])
}

bter.res.all <- bind_rows(bter.res[1:3])

id <- read.table("input/Gene_Info/Bter_IDwDesc.tsv", 
              sep = "\t", quote = "", header = T)


for (i in 1:nrow(bter.res.all)){
  d <- unique(id$Description[id$GeneID == bter.res.all$Gene[i]])
  if(length(d) > 1){
    bter.res.all$Desc[i] <- d[1]
  } else {
    bter.res.all$Desc[i] <- d  
  }
}

write.table(bter.res.all, paste(wd, "Bter_DeSeqResults_All.tsv", sep = ""),
            col.names = T, row.names = F, quote = F, sep = "\t")

bter.res.sig <- subset(bter.res.all, padj < 0.1)

write.table(bter.res.sig, paste(wd, "Bter_DeSeqResults_FDRlt0.1.tsv", sep =""), 
            col.names = T, row.names = F, quote = F, sep = "\t")
```

Genes of interest?

I'm going to take the top 10 most significant and top 10 biggest logFC per treatment. 
Genes that overlap between two (and between the three treatments) will be of particular interest

```{r}
nona <- bter.res.all[!is.na(bter.res.all$padj),]
nona$LogDif <- sub('-', '', nona$log2FoldChange)
tmp <- nona[nona$Contrast == treats[1],]
sig.wound <- head(tmp$Gene[order(tmp$padj)], n = 10)
top.wound <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)
tmp <- nona[nona$Contrast == treats[2],]
sig.pos <- head(tmp$Gene[order(tmp$padj)], n = 10)
top.pos <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)
tmp <- nona[nona$Contrast == treats[3],]
sig.neg <-  head(tmp$Gene[order(tmp$padj)], n = 10)
top.neg <- head(tmp$Gene[order(tmp$LogDif, decreasing = T)], n = 10)

goi <- c(sig.wound, top.wound, sig.pos, top.pos, sig.neg, top.neg)
stat <- c(rep("Top 10 Sig Wound", 10), rep("Top 10 log2FC Wound", 10),
          rep("Top 10 Sig Gram Positive", 10), rep("Top 10 log2FC Gram Positive", 10),
          rep("Top 10 Sig Gram Negative", 10), rep("Top 10 log2FC Gram Negative", 10))

goi.df <- as.data.frame(cbind(goi, stat))


goi.wound <- unique(c(sig.wound, top.wound))
goi.pos <- unique(c(sig.pos, top.pos))
goi.neg <- unique(c(sig.neg, top.neg))
```

###Visualise: Venn Diagrams

```{r}
top.sig <- list("Wound" = sig.wound,
                 "Gram Positive" = sig.pos,
                 "Gram Negative" = sig.neg)

p <- ggvenn(top.sig,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Top 10 Genes per Treatment by adjPvalue")
```

```{r}
dir.create(paste(wd, "Plots/", sep = ""))
ggsave(paste(wd, "Plots/Bter_TopAdjPvalue_Venn.pdf", sep = ""))
```

```{r}
top.lfc <- list("Wound" = top.wound,
                 "Gram Positive" = top.pos,
                 "Gram Negative" = top.neg)

p <- ggvenn(top.lfc,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Top 10 Genes per Treatment by log2FC")
```



```{r}
ggsave(paste(wd, "Plots/Bter_Toplog2FC_Venn.pdf", sep = ""))
```

```{r}
goi <- list("Wound" = goi.wound,
                 "Gram Positive" = goi.pos,
                 "Gram Negative" = goi.neg)

p <- ggvenn(goi,
       set_name_size = 5,
       show_percentage = F,
       fill_color = c("yellow", "blue", "green"))
p +  ggtitle("Genes of Interest Across Treatments")
```


```{r}
ggsave(paste(wd, "Plots/Bter_GOI_Venn.pdf", sep = ""))
```

```{r}
goi.mat <- makeMatrix(goi.df)
goi.mat
```

```{r}
goi.mat$Gene[goi.mat$AllGoi == 1]
```

hymenoptaecin, hymenoptaecin-like, apidaecins type 73, protein Malvolio, uncharacterised LOC125385027, KATNAL2, Defensin

```{r}
goi.mat$Gene[goi.mat$BacterialGoi == 1]
```

regulator of microtubule dynamics protein 1-like

```{r}
goi.mat$Gene[goi.mat$AllSig == 1]
```

hymenoptaecin, hymenoptaecin-like, apidaecins type 73, protein Malvolio

```{r}
goi.mat$Gene[goi.mat$AllTop == 1]
```

uncharacterised LOC125385027, KATNAL2, Defensin

###Visualise: Expression BoxPlots

```{r}
dir.create(paste(wd, "Plots/ExpressionPlots/", sep = ""))
```

####Significant Wound

```{r}
goi.mat$Gene[goi.mat$SigWound == 1]
genes <- goi.mat$Gene[goi.mat$SigWound == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = bter.dds)
}
```

All Wound: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = bter.dds, 
          filename = "WoundSig")
```


####TopFC Wound

```{r}
goi.mat$Gene[goi.mat$TopWound == 1]
genes <- goi.mat$Gene[goi.mat$TopWound == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = bter.dds)
}
```

All Wound: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = bter.dds, 
          filename = "Woundlog2FC")
```


####Significant GramPos


```{r}
goi.mat$Gene[goi.mat$SigPos == 1]
```


```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = bter.dds)
}
```


All Gram Pos Sig: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = bter.dds, 
          filename = "PosSig")
```


####TopFC GramPos


```{r}
goi.mat$Gene[goi.mat$TopPos == 1]
genes <- goi.mat$Gene[goi.mat$TopPos == 1]
```



```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = bter.dds)
}
```


All log Gram Pos: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = bter.dds, 
          filename = "Poslog2FC")
```

####Significant GramNeg


```{r}
goi.mat$Gene[goi.mat$SigNeg == 1]
genes <- goi.mat$Gene[goi.mat$SigNeg == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = bter.dds)
}
```

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = bter.dds, 
          filename = "NegSig")
```


####TopFC GramPos


```{r}
goi.mat$Gene[goi.mat$TopNeg == 1]
genes <- goi.mat$Gene[goi.mat$TopNeg == 1]
```

```{r}
for (i in 1:length(genes)){
  visiGene(gene = genes[i], genename = genes[i], dds = bter.dds)
}
```

All log Gram Neg: 

```{r}
visiFacet(genes = genes,
          genenames = genes,
          dds = bter.dds, 
          filename = "Neglog2FC")
```


###Visualise: PCA

Redo after removing outliers.

```{r}
vsd <- vst(bter.dds, blind = F)

bter.pca <- plotPCA(vsd, intgroup="Treatment", returnData = TRUE)

ggplot(bter.pca, aes(x = PC1, y = PC2, color = Treatment)) +
  geom_point() 
```

```{r}
ggsave(paste(wd, "Plots/Bter_PCA_Simple.pdf", sep = ""))
```

```{r}
ggplot(bter.pca, aes(x = PC1, y = PC2, color = Treatment)) +
  geom_point() +
  geom_label_repel(
    data = bter.pca,
    aes(label = name),
    size = 3
  )
```
```{r}
ggsave(paste(wd, "Plots/Bter_PCA_Labelled.pdf", sep = ""))
```

```{r}
bter.pca$Treat2 <- c(rep("Naive", 2), rep("Treatment", 6))

ggplot(bter.pca, aes(x = PC1, y = PC2, color = Treat2)) +
  geom_point() +
  stat_ellipse()
```

```{r}
ggsave(paste(wd, "Plots/Bter_PCA_TrtVUnTrt.pdf", sep = ""))
```

###Visualise: Smears

Amel/Bter consideration : there is a shared gene name that causes issues. To fix:

```{r}
bter.res.all$Gene[bter.res.all$Gene == "Kr-h1"] <- "Kr-h1_Bter"

ortho$GeneID[ortho$GeneID == "Kr-h1" & ortho$Species == "Bter"] <- "Kr-h1_Bter"
```

```{r}
plot.df <- makeMetaDf(bter.res.all)
plot.df %>%
  filter(Condition == "Wound") %>%
  arrange(padj)
```



```{r}
plot.df$Condition <- factor(plot.df$Condition, levels = c("Wound",
                                                          "Gram Positive",
                                                          "Gram Negative"))

ggplot(plot.df, aes(x = log(baseMean), y = log2FoldChange)) +
  geom_point(aes(colour = Sig)) +
  facet_grid(~Condition) +
  scale_color_manual(values=c("black", "Red"), guide = NULL) +
  theme(legend.position = "bottom")
```



```{r}
ggsave(paste(wd, "Plots/Bter_SmearPlot.pdf", sep = ""))
```


###Visualise: Rainclouds

```{r}
sig.genes <- unique(plot.df$Gene[plot.df$padj < 0.1])
plot.sig <- plot.df[plot.df$Gene %in% sig.genes,]

to.add.naive <- plot.sig[plot.sig$Gene %in% sig.genes &
                               plot.sig$Condition == "Wound",]
to.add.naive$Sig <- "No"
to.add.naive$Condition <- "Naive"
to.add.naive$log2FoldChange <- 0

plot.sig <- rbind(plot.sig, to.add.naive)

plot.sig$Condition <- factor(plot.sig$Condition, levels = c("Naive",
                                                            "Wound", 
                                                            "Gram Positive", 
                                                            "Gram Negative"))

plot.sig$Significance[plot.sig$padj > 0.1] <- "Not Significant"
plot.sig$Significance[plot.sig$padj < 0.1 & plot.sig$padj > 0.05] <- 
  "Significant (FDR < 0.1)"
plot.sig$Significance[plot.sig$padj < 0.05 & plot.sig$padj > 0.01] <-
  "Significant (FDR < 0.05)"
plot.sig$Significance[plot.sig$padj < 0.01 & plot.sig$padj > 0.001] <-
  "Significant (FDR < 0.01)"
plot.sig$Significance[plot.sig$padj < 0.001] <-
  "Significant (FDR < 0.001)"

plot.sig$Significance <- factor(plot.sig$Significance, levels =
                                 c("Not Significant",
                                   "Significant (FDR < 0.1)",
                                   "Significant (FDR < 0.05)",
                                   "Significant (FDR < 0.01)",
                                   "Significant (FDR < 0.001)"))


plot.sig$Function <- factor(plot.sig$Function, levels = c("Immune",
                                                          "Putative Immune", 
                                                          "None Immune", 
                                                          "Unknown"))

ggplot(plot.sig[!plot.sig$Condition == "Naive",], 
       (aes(x = Condition, y = log2FoldChange))) +
  geom_jitter(aes(colour = Function, shape = Significance)) +
  scale_colour_manual(values = c("#2c8cd9", "#7a2cd9", "darkgrey", "plum")) +
  scale_shape_manual(values = c(1, 5, 18, 19, 17)) 
```

```{r}
ggsave(paste(wd, "Plots/Bter_logFCJitter.pdf", sep = ""))
```

```{r}
ggplot(plot.sig, 
       (aes(x = Condition, y = log2FoldChange))) +
  geom_point(aes(colour = Function, shape = Significance)) +
  geom_line( linetype = "dashed", alpha = 0.2, aes(group = Gene, colour = Function)) +
  scale_colour_manual(values = c("#2c8cd9", "#7a2cd9", "darkgrey", "plum")) +
  scale_shape_manual(values = c(1, 5, 18, 19, 17)) 
```

```{r}
ggsave(paste(wd, "Plots/Bter_logFCScatter_GeneLine.pdf", sep = ""))
```

###Visualise: Volcano Plots

```{r}
plot.df$Function <- factor(plot.df$Function, levels = c("Immune",
                                                        "Putative Immune",
                                                        "None Immune",
                                                        "Unknown"))

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.1", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Bter_Simple_FDR0.1_Volcano.pdf", sep = ""))
```


```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.1", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```
```{r}
ggsave(paste(wd, "Plots/Bter_Class_FDR0.1_Volcano.pdf", sep = ""))
```

```{r}
plot.df$Direction2 <- "Not Significant"
plot.df$Direction2[plot.df$padj < 0.05 &
                    plot.df$log2FoldChange > 0 ] <- "Up-Regulated"
plot.df$Direction2[plot.df$padj < 0.05 &
                    plot.df$log2FoldChange < 0 ] <- "Down-Regulated"

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction2), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.05", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```

```{r}
ggsave(paste(wd, "Plots/Bter_Simple_FDR0.05_Volcano.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction2, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.05", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Bter_Class_FDR0.05_Volcano.pdf", sep = ""))
```


```{r}
plot.df$Direction3 <- "Not Significant"
plot.df$Direction3[plot.df$padj < 0.001 &
                    plot.df$log2FoldChange > 0 ] <- "Up-Regulated"
plot.df$Direction3[plot.df$padj < 0.001 &
                    plot.df$log2FoldChange < 0 ] <- "Down-Regulated"

ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Bter_Simple_FDR0.001_Volcano.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") 
```


```{r}
ggsave(paste(wd, "Plots/Bter_Class_FDR0.001_Volcano.pdf", sep = ""))
```

Label top 3 sig genes per treatment

```{r}
sig.wound[1:3]
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "LOC100631061"] <- "Hymenoptaecin"
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "LOC125385347"] <- "Hymenoptaecin-like"
plot.df$Label[plot.df$Condition == "Wound" & 
                plot.df$Gene == "LOC100649281"] <- "Maltase A1-like"

sig.pos[1:3]
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "LOC100631061"] <- "Hymenoptaecin"
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "LOC100644101"] <- "Uncharacterised LOC100644101"
plot.df$Label[plot.df$Condition == "Gram Positive" & 
                plot.df$Gene == "LOC100649867"] <- "Apidaecins type 73"

sig.neg[1:3]
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "LOC100631061"] <- "Hymenoptaecin"
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "LOC125385347"] <- "Hymenoptaecin-like"
plot.df$Label[plot.df$Condition == "Gram Negative" & 
                plot.df$Gene == "LOC100649867"] <- "Apidaecins type 73"
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3), size = 1, 
             position = position_jitter(w=0.1, h = 0.1)) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 3)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") +
  geom_label_repel(
    data = plot.df[!is.na(plot.df$Label),],
    size = 2,
    aes(label = Label),
    point.padding = unit(0.3, "lines"),
    box.padding = 1
  ) 

```


```{r}
ggsave(paste(wd, "Plots/Bter_Simple_FDR0.001_Volcano_top3Label.pdf", sep = ""))
```

```{r}
ggplot(plot.df,
     aes(x = log2FoldChange, y = Neglog10adjPvalue)) +
  geom_point(aes(colour = Direction3, shape = Function), size = 1) +
  scale_colour_manual(values=c("Up-Regulated" = "red",
                               "Not Top" = "black",  
                               "Down-Regulated" = "blue" ),
                      breaks = c("Up-Regulated", "Down-Regulated")) +
  scale_shape_manual(values= c(19, 17, 1, 3)) +
  guides(colour = guide_legend("FDR < 0.001", nrow = 2),
         shape = guide_legend("Gene Function", nrow = 4)) +
  ylab("-log10(adjPvalue)") +
  theme(legend.position = "bottom") +
    facet_grid(~Condition, scales = "free_y") +
  geom_label_repel(
    data = plot.df[!is.na(plot.df$Label),],
    size = 2,
    aes(label = Label),
    point.padding = unit(0.3, "lines"),
    box.padding = 1
  ) 

```

```{r}
ggsave(paste(wd, "Plots/Bter_Class_FDR0.001_Volcano_top3Label.pdf", sep = ""))
```

Finish with writing the new ortho table (with a caveat in case I've had to rerun this project)

```{r}
ortho <- ortho[!duplicated(ortho),]
write.table(ortho, "input/Ortho_Files/OrthoFinder_Combined_Parsed_Jul22.tsv",
            col.names = T, row.names = F, quote = F, sep = "\t")
```

##SessionInfo

```{r}
dir.create("SessionLogs/")
writeLines(capture.output(sessionInfo()),
           "SessionLogs/Proj_Pt1_SesInfo.txt")
sessionInfo()
```



